// @qwik-ui/headless/components/carousel/use/use-items.qwik.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const qwik = require("@builder.io/qwik");
const utils = require("./utils.qwik.cjs");
const useVisible = require("./use-visible.qwik.cjs");
const useItems = (ref, params) => {
  const visible = useVisible.useVisible(ref);
  const loop = params?.loop ?? true;
  const transition = params?.transition || 350;
  const isFirst = params?.startAt && params?.startAt === 0 || false;
  const isLast = params?.startAt && params?.startAt + 1 === utils.getCount(ref) || false;
  const index = params?.startAt || 0;
  const total = qwik.useSignal(0);
  const active = qwik.useStore({
    isFirst: qwik.useSignal(isFirst),
    isLast: qwik.useSignal(isLast),
    current: qwik.useSignal(index)
  });
  const scrollAt = /* @__PURE__ */ qwik.inlinedQrl((index2) => {
    const [active2, ref2, transition2] = qwik.useLexicalScope();
    const count = utils.getCount(ref2);
    const element = utils.getElement(ref2, index2);
    if (!ref2?.value || !count) {
      console.warn(`Can't jump to ${index2} because the carousel elements is empty.`);
      return;
    }
    if (!element) {
      console.warn(`Can't jump to ${index2} because the element index ${index2} doesn't exist.`);
      return;
    }
    setTimeout(() => element.scrollIntoView({
      behavior: "smooth",
      block: "nearest",
      inline: "center"
    }), transition2);
    active2.current.value = index2;
  }, "useItems_scrollAt_w0rF5B4LQ7s", [
    active,
    ref,
    transition
  ]);
  const previous = /* @__PURE__ */ qwik.inlinedQrl(() => {
    const [active2, loop2, ref2, scrollAt2] = qwik.useLexicalScope();
    if (!loop2 && active2.isFirst.value)
      return;
    const index2 = active2.isFirst.value ? utils.getCount(ref2) - 1 : active2.current.value - 1;
    scrollAt2(index2);
  }, "useItems_previous_4tOoNm3D6To", [
    active,
    loop,
    ref,
    scrollAt
  ]);
  const next = /* @__PURE__ */ qwik.inlinedQrl(() => {
    const [active2, loop2, ref2, scrollAt2] = qwik.useLexicalScope();
    const max = utils.getCount(ref2) - 1;
    if (!loop2 && active2.current.value === max)
      return;
    const index2 = active2.current.value === max ? 0 : active2.current.value + 1;
    scrollAt2(index2);
  }, "useItems_next_Aw4EtPIWJ2M", [
    active,
    loop,
    ref,
    scrollAt
  ]);
  qwik.useOnDocument("focus", /* @__PURE__ */ qwik.inlinedQrl(() => {
    const [active2, ref2] = qwik.useLexicalScope();
    utils.getElements(ref2).forEach((element, index2) => {
      if (!element.contains(document.activeElement))
        return;
      active2.current.value = index2;
    });
  }, "useItems_useOnDocument_o3Qy4RtDcX4", [
    active,
    ref
  ]));
  qwik.useTaskQrl(/* @__PURE__ */ qwik.inlinedQrl(({ track }) => {
    const [active2, ref2] = qwik.useLexicalScope();
    const index2 = track(() => active2.current.value);
    active2.isFirst.value = index2 === 0;
    active2.isLast.value = index2 + 1 === utils.getCount(ref2);
  }, "useItems_useTask_koFKDjM01VA", [
    active,
    ref
  ]));
  qwik.useVisibleTaskQrl(/* @__PURE__ */ qwik.inlinedQrl(() => {
    const [active2, ref2, scrollAt2, total2] = qwik.useLexicalScope();
    total2.value = utils.getCount(ref2);
    scrollAt2(active2.current.value);
  }, "useItems_useVisibleTask_FPaY6NnxHbA", [
    active,
    ref,
    scrollAt,
    total
  ]));
  return {
    active,
    visible,
    total,
    next,
    previous,
    scrollAt
  };
};
exports.useItems = useItems;
