// @qwik-ui/headless/components/collapse/collapse.qwik.mjs
import { componentQrl, inlinedQrl, useSignal, _jsxQ, _fnSignal, useLexicalScope, _jsxC, Slot, _IMMUTABLE } from "@builder.io/qwik";
const Collapse = /* @__PURE__ */ componentQrl(/* @__PURE__ */ inlinedQrl((props) => {
  const random = Math.random() * 1e3;
  const state = useSignal("closed");
  return /* @__PURE__ */ _jsxQ("div", null, {
    class: _fnSignal((p0) => p0.class, [
      props
    ], "p0.class"),
    "data-state": _fnSignal((p0) => p0.value, [
      state
    ], "p0.value")
  }, [
    /* @__PURE__ */ _jsxQ("style", null, {
      dangerouslySetInnerHTML: `
      @keyframes DummyIn {
        to: { opacity(1); }
      }
      @keyframes DummyOut {
        to: { opacity(0); }
      }
      [data-state='closing'] > div {
        animation: DummyOut 1ms linear forwards; 
      }
      [data-state='open'] > div {
        animation: DummyIn 1ms linear forwards; 
      }
      [data-state='closed'] > div {
        display: none;
      }
    `
    }, null, 3, null),
    /* @__PURE__ */ _jsxQ("button", {
      "aria-controls": random.toString()
    }, {
      "aria-expanded": _fnSignal((p0) => p0.value === "open", [
        state
      ], 'p0.value==="open"'),
      onClick$: /* @__PURE__ */ inlinedQrl(() => {
        const [state2] = useLexicalScope();
        return state2.value === "open" ? state2.value = "closing" : state2.value = "open";
      }, "Collapse_component_div_button_onClick_E5blXWvrWRA", [
        state
      ])
    }, /* @__PURE__ */ _jsxC(Slot, {
      name: "label",
      [_IMMUTABLE]: {
        name: _IMMUTABLE
      }
    }, 3, "Ji_0"), 1, null),
    /* @__PURE__ */ _jsxQ("div", {
      id: random.toString()
    }, {
      onAnimationEnd$: /* @__PURE__ */ inlinedQrl(() => {
        const [state2] = useLexicalScope();
        state2.value === "closing" ? state2.value = "closed" : state2.value;
      }, "Collapse_component_div_div_onAnimationEnd_hh05FvQ4qm8", [
        state
      ])
    }, /* @__PURE__ */ _jsxC(Slot, {
      name: "content",
      [_IMMUTABLE]: {
        name: _IMMUTABLE
      }
    }, 3, "Ji_1"), 1, null)
  ], 1, "Ji_2");
}, "Collapse_component_YdCJfq0QZV8"));
export {
  Collapse
};
