// @qwik-ui/headless/components/collapse/collapse.qwik.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const qwik = require("@builder.io/qwik");
const Collapse = /* @__PURE__ */ qwik.componentQrl(/* @__PURE__ */ qwik.inlinedQrl((props) => {
  const random = Math.random() * 1e3;
  const state = qwik.useSignal("closed");
  return /* @__PURE__ */ qwik._jsxQ("div", null, {
    class: qwik._fnSignal((p0) => p0.class, [
      props
    ], "p0.class"),
    "data-state": qwik._fnSignal((p0) => p0.value, [
      state
    ], "p0.value")
  }, [
    /* @__PURE__ */ qwik._jsxQ("style", null, {
      dangerouslySetInnerHTML: `
      @keyframes DummyIn {
        to: { opacity(1); }
      }
      @keyframes DummyOut {
        to: { opacity(0); }
      }
      [data-state='closing'] > div {
        animation: DummyOut 1ms linear forwards; 
      }
      [data-state='open'] > div {
        animation: DummyIn 1ms linear forwards; 
      }
      [data-state='closed'] > div {
        display: none;
      }
    `
    }, null, 3, null),
    /* @__PURE__ */ qwik._jsxQ("button", {
      "aria-controls": random.toString()
    }, {
      "aria-expanded": qwik._fnSignal((p0) => p0.value === "open", [
        state
      ], 'p0.value==="open"'),
      onClick$: /* @__PURE__ */ qwik.inlinedQrl(() => {
        const [state2] = qwik.useLexicalScope();
        return state2.value === "open" ? state2.value = "closing" : state2.value = "open";
      }, "Collapse_component_div_button_onClick_E5blXWvrWRA", [
        state
      ])
    }, /* @__PURE__ */ qwik._jsxC(qwik.Slot, {
      name: "label",
      [qwik._IMMUTABLE]: {
        name: qwik._IMMUTABLE
      }
    }, 3, "Ji_0"), 1, null),
    /* @__PURE__ */ qwik._jsxQ("div", {
      id: random.toString()
    }, {
      onAnimationEnd$: /* @__PURE__ */ qwik.inlinedQrl(() => {
        const [state2] = qwik.useLexicalScope();
        state2.value === "closing" ? state2.value = "closed" : state2.value;
      }, "Collapse_component_div_div_onAnimationEnd_hh05FvQ4qm8", [
        state
      ])
    }, /* @__PURE__ */ qwik._jsxC(qwik.Slot, {
      name: "content",
      [qwik._IMMUTABLE]: {
        name: qwik._IMMUTABLE
      }
    }, 3, "Ji_1"), 1, null)
  ], 1, "Ji_2");
}, "Collapse_component_YdCJfq0QZV8"));
exports.Collapse = Collapse;
