// @qwik-ui/headless/components/rating/rating.qwik.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const qwik = require("@builder.io/qwik");
const jsxRuntime = require("@builder.io/qwik/jsx-runtime");
const rating = require("./rating.css.qwik.cjs");
const Rating = /* @__PURE__ */ qwik.componentQrl(/* @__PURE__ */ qwik.inlinedQrl((props) => {
  const uniqueId = qwik.useId();
  qwik.useStylesQrl(/* @__PURE__ */ qwik.inlinedQrl(rating, "Rating_component_useStyles_5CPM8lFif0o"));
  const state = qwik.useStore({
    index: 0,
    tempIndex: 0
  });
  const onItemClick$ = /* @__PURE__ */ qwik.inlinedQrl((index) => {
    const [props2, state2] = qwik.useLexicalScope();
    state2.index = index;
    if (props2.onChange$)
      props2.onChange$(index + 1);
  }, "Rating_component_onItemClick_Kgg0ONfmorA", [
    props,
    state
  ]);
  return /* @__PURE__ */ qwik._jsxQ("span", null, {
    class: "wrapper"
  }, Array.from({
    length: props.total ?? 5
  }, (_, index) => index).map((item, i) => {
    const Icon = props.icon || DefaultIcon;
    return /* @__PURE__ */ qwik._jsxC(RatingIcon, {
      index: i,
      name: `rating-${uniqueId}`,
      onChange$: /* @__PURE__ */ qwik.inlinedQrl(() => {
        const [i2, onItemClick$2] = qwik.useLexicalScope();
        return onItemClick$2(i2);
      }, "Rating_component_span_RatingIcon_onChange_QA052QVeZ8A", [
        i,
        onItemClick$
      ]),
      get value() {
        return props.value;
      },
      children: /* @__PURE__ */ qwik._jsxC(Icon, null, 3, "wv_0"),
      [qwik._IMMUTABLE]: {
        name: qwik._IMMUTABLE,
        value: qwik._fnSignal((p0) => p0.value, [
          props
        ], "p0.value")
      }
    }, 1, item);
  }), 1, "wv_1");
}, "Rating_component_x6Va0MlPc0o"));
const RatingIcon = /* @__PURE__ */ qwik.componentQrl(/* @__PURE__ */ qwik.inlinedQrl((props) => {
  return /* @__PURE__ */ qwik._jsxC(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ qwik._jsxQ("label", null, {
        for: qwik._fnSignal((p0) => `${p0.name}-${p0.index}`, [
          props
        ], "`${p0.name}-${p0.index}`")
      }, /* @__PURE__ */ qwik._jsxC(qwik.Slot, null, 3, "wv_2"), 1, null),
      /* @__PURE__ */ qwik._jsxQ("input", {
        onChange$: props.onChange$
      }, {
        checked: qwik._fnSignal((p0) => p0.value === p0.index + 1, [
          props
        ], "p0.value===p0.index+1"),
        hidden: true,
        id: qwik._fnSignal((p0) => `${p0.name}-${p0.index}`, [
          props
        ], "`${p0.name}-${p0.index}`"),
        name: qwik._fnSignal((p0) => p0.name, [
          props
        ], "p0.name"),
        type: "radio"
      }, null, 2, null)
    ]
  }, 1, "wv_3");
}, "RatingIcon_component_W7soSSdH5C0"));
const DefaultIcon = /* @__PURE__ */ qwik.componentQrl(/* @__PURE__ */ qwik.inlinedQrl(() => /* @__PURE__ */ qwik._jsxC(jsxRuntime.Fragment, {
  children: "⭐️"
}, 3, "wv_4"), "DefaultIcon_component_Gtj0zavuMg8"));
exports.DefaultIcon = DefaultIcon;
exports.Rating = Rating;
exports.RatingIcon = RatingIcon;
