// @qwik-ui/headless/components/rating/rating.qwik.mjs
import { componentQrl, inlinedQrl, useId, useStylesQrl, useStore, _jsxQ, _jsxC, useLexicalScope, _IMMUTABLE, _fnSignal, Slot } from "@builder.io/qwik";
import { Fragment } from "@builder.io/qwik/jsx-runtime";
import styles from "./rating.css.qwik.mjs";
const Rating = /* @__PURE__ */ componentQrl(/* @__PURE__ */ inlinedQrl((props) => {
  const uniqueId = useId();
  useStylesQrl(/* @__PURE__ */ inlinedQrl(styles, "Rating_component_useStyles_5CPM8lFif0o"));
  const state = useStore({
    index: 0,
    tempIndex: 0
  });
  const onItemClick$ = /* @__PURE__ */ inlinedQrl((index) => {
    const [props2, state2] = useLexicalScope();
    state2.index = index;
    if (props2.onChange$)
      props2.onChange$(index + 1);
  }, "Rating_component_onItemClick_Kgg0ONfmorA", [
    props,
    state
  ]);
  return /* @__PURE__ */ _jsxQ("span", null, {
    class: "wrapper"
  }, Array.from({
    length: props.total ?? 5
  }, (_, index) => index).map((item, i) => {
    const Icon = props.icon || DefaultIcon;
    return /* @__PURE__ */ _jsxC(RatingIcon, {
      index: i,
      name: `rating-${uniqueId}`,
      onChange$: /* @__PURE__ */ inlinedQrl(() => {
        const [i2, onItemClick$2] = useLexicalScope();
        return onItemClick$2(i2);
      }, "Rating_component_span_RatingIcon_onChange_QA052QVeZ8A", [
        i,
        onItemClick$
      ]),
      get value() {
        return props.value;
      },
      children: /* @__PURE__ */ _jsxC(Icon, null, 3, "wv_0"),
      [_IMMUTABLE]: {
        name: _IMMUTABLE,
        value: _fnSignal((p0) => p0.value, [
          props
        ], "p0.value")
      }
    }, 1, item);
  }), 1, "wv_1");
}, "Rating_component_x6Va0MlPc0o"));
const RatingIcon = /* @__PURE__ */ componentQrl(/* @__PURE__ */ inlinedQrl((props) => {
  return /* @__PURE__ */ _jsxC(Fragment, {
    children: [
      /* @__PURE__ */ _jsxQ("label", null, {
        for: _fnSignal((p0) => `${p0.name}-${p0.index}`, [
          props
        ], "`${p0.name}-${p0.index}`")
      }, /* @__PURE__ */ _jsxC(Slot, null, 3, "wv_2"), 1, null),
      /* @__PURE__ */ _jsxQ("input", {
        onChange$: props.onChange$
      }, {
        checked: _fnSignal((p0) => p0.value === p0.index + 1, [
          props
        ], "p0.value===p0.index+1"),
        hidden: true,
        id: _fnSignal((p0) => `${p0.name}-${p0.index}`, [
          props
        ], "`${p0.name}-${p0.index}`"),
        name: _fnSignal((p0) => p0.name, [
          props
        ], "p0.name"),
        type: "radio"
      }, null, 2, null)
    ]
  }, 1, "wv_3");
}, "RatingIcon_component_W7soSSdH5C0"));
const DefaultIcon = /* @__PURE__ */ componentQrl(/* @__PURE__ */ inlinedQrl(() => /* @__PURE__ */ _jsxC(Fragment, {
  children: "⭐️"
}, 3, "wv_4"), "DefaultIcon_component_Gtj0zavuMg8"));
export {
  DefaultIcon,
  Rating,
  RatingIcon
};
