// @qwik-ui/headless/components/autocomplete/autocomplete-trigger.qwik.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const qwik = require("@builder.io/qwik");
const autocompleteContextId = require("./autocomplete-context-id.qwik.cjs");
const AutocompleteTrigger = /* @__PURE__ */ qwik.componentQrl(/* @__PURE__ */ qwik.inlinedQrl(({ ...props }) => {
  const contextService = qwik.useContext(autocompleteContextId);
  const triggerRefSig = qwik.useSignal();
  qwik.useVisibleTaskQrl(/* @__PURE__ */ qwik.inlinedQrl(function registerInputRefTask() {
    const [contextService2, triggerRefSig2] = qwik.useLexicalScope();
    contextService2.triggerRefSig.value = triggerRefSig2.value;
  }, "AutocompleteTrigger_component_useVisibleTask_WMHKcaf20DE", [
    contextService,
    triggerRefSig
  ]));
  return /* @__PURE__ */ qwik._jsxS("button", {
    ...props,
    children: /* @__PURE__ */ qwik._jsxC(qwik.Slot, null, 3, "h3_0"),
    // add their own custom onClick with our onClick functionality
    onClick$: [
      /* @__PURE__ */ qwik.inlinedQrl(() => {
        const [contextService2] = qwik.useLexicalScope();
        contextService2.isTriggerExpandedSig.value = !contextService2.isTriggerExpandedSig.value;
      }, "AutocompleteTrigger_component_button_onClick_Z5Ur56odKNw", [
        contextService
      ]),
      props.onClick$
    ],
    ref: triggerRefSig
  }, {
    "aria-controls": qwik._fnSignal((p0) => p0.listBoxId, [
      contextService
    ], "p0.listBoxId"),
    "aria-expanded": qwik._fnSignal((p0) => p0.isTriggerExpandedSig.value, [
      contextService
    ], "p0.isTriggerExpandedSig.value"),
    "aria-haspopup": "listbox",
    "aria-label": "listbox toggle trigger",
    id: qwik._fnSignal((p0) => p0.triggerId, [
      contextService
    ], "p0.triggerId"),
    tabIndex: -1
  }, 0, "h3_1");
}, "AutocompleteTrigger_component_X4cPs5Kv8iA"));
exports.AutocompleteTrigger = AutocompleteTrigger;
