// @qwik-ui/headless/components/autocomplete/autocomplete-listbox.qwik.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const qwik = require("@builder.io/qwik");
const autocompleteContextId = require("./autocomplete-context-id.qwik.cjs");
const keyCode_type = require("../../utils/key-code.type.qwik.cjs");
const preventedKeys = [
  keyCode_type.KeyCode.Home,
  keyCode_type.KeyCode.End,
  keyCode_type.KeyCode.PageDown,
  keyCode_type.KeyCode.PageUp,
  keyCode_type.KeyCode.ArrowDown,
  keyCode_type.KeyCode.ArrowUp
];
const AutocompleteListbox = /* @__PURE__ */ qwik.componentQrl(/* @__PURE__ */ qwik.inlinedQrl((props) => {
  const listboxRefSig = qwik.useSignal();
  const contextService = qwik.useContext(autocompleteContextId);
  qwik.useVisibleTaskQrl(/* @__PURE__ */ qwik.inlinedQrl(function preventDefaultTask({ cleanup }) {
    const [listboxRefSig2] = qwik.useLexicalScope();
    function keyHandler(e) {
      if (preventedKeys.includes(e.key))
        e.preventDefault();
    }
    listboxRefSig2.value?.addEventListener("keydown", keyHandler);
    cleanup(() => {
      listboxRefSig2.value?.removeEventListener("keydown", keyHandler);
    });
  }, "AutocompleteListbox_component_useVisibleTask_Ezcf8dU18EU", [
    listboxRefSig
  ]));
  qwik.useVisibleTaskQrl(/* @__PURE__ */ qwik.inlinedQrl(function focusListboxTask({ track }) {
    const [contextService2, listboxRefSig2] = qwik.useLexicalScope();
    if (track(() => contextService2.isTriggerExpandedSig.value))
      listboxRefSig2.value?.focus();
  }, "AutocompleteListbox_component_useVisibleTask_1_Gv0Ci5GgdFc", [
    contextService,
    listboxRefSig
  ]));
  qwik.useVisibleTaskQrl(/* @__PURE__ */ qwik.inlinedQrl(function registerListboxRefTask() {
    const [contextService2, listboxRefSig2] = qwik.useLexicalScope();
    contextService2.listBoxRefSig.value = listboxRefSig2.value;
  }, "AutocompleteListbox_component_useVisibleTask_2_z4a2kcjy0s0", [
    contextService,
    listboxRefSig
  ]));
  return /* @__PURE__ */ qwik._jsxS("ul", {
    ref: listboxRefSig,
    ...props,
    children: /* @__PURE__ */ qwik._jsxC(qwik.Slot, null, 3, "D6_0"),
    // aria-label={!contextService.labelRef.value ? contextService.inputValue.value : undefined}
    onKeyDown$: [
      /* @__PURE__ */ qwik.inlinedQrl((e) => {
        const [contextService2] = qwik.useLexicalScope();
        const availableOptions = contextService2.filteredOptionsStore.map((option) => option.value);
        const target = e.target;
        const currentIndex = availableOptions.indexOf(target);
        if (e.key === "ArrowDown") {
          if (currentIndex === availableOptions.length - 1)
            availableOptions[0]?.focus();
          else
            availableOptions[currentIndex + 1]?.focus();
        }
        if (e.key === "ArrowUp") {
          if (currentIndex <= 0)
            availableOptions[availableOptions.length - 1]?.focus();
          else
            availableOptions[currentIndex - 1]?.focus();
        }
        if (e.key === "Home")
          availableOptions[0]?.focus();
        if (e.key === "End")
          availableOptions[availableOptions.length - 1]?.focus();
      }, "AutocompleteListbox_component_ul_onKeyDown_IqVk5yUZAFQ", [
        contextService
      ]),
      props.onKeyDown$
    ]
  }, {
    id: qwik._fnSignal((p0) => p0.listBoxId, [
      contextService
    ], "p0.listBoxId"),
    role: "listbox",
    style: qwik._fnSignal((p0, p1) => `
          display: ${p0.isTriggerExpandedSig.value ? "block" : "none"}; position: absolute; z-index: 1; ${p1.style}
      `, [
      contextService,
      props
    ], '`\n          display: ${p0.isTriggerExpandedSig.value?"block":"none"}; position: absolute; z-index: 1; ${p1.style}\n      `')
  }, 0, "D6_1");
}, "AutocompleteListbox_component_GKhRxpVEBKY"));
exports.AutocompleteListbox = AutocompleteListbox;
exports._auto_preventedKeys = preventedKeys;
