// @qwik-ui/headless/components/autocomplete/autocomplete-root.qwik.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const qwik = require("@builder.io/qwik");
const dom = require("@floating-ui/dom");
const autocompleteContextId = require("./autocomplete-context-id.qwik.cjs");
const AutocompleteRoot = /* @__PURE__ */ qwik.componentQrl(/* @__PURE__ */ qwik.inlinedQrl((props) => {
  const props1 = qwik._restProps(props, [
    "defaultValue"
  ]);
  qwik.useVisibleTaskQrl(/* @__PURE__ */ qwik.inlinedQrl(function deprecatedComponentError() {
    throw new Error("Qwik UI: The Autocomplete component has been deprecated in favor of the Combobox. Please use the Combobox component instead.");
  }, "AutocompleteRoot_component_useVisibleTask_YM0YQy6ylMI"));
  const optionsStore = qwik.useStore([]);
  const filteredOptionsStore = qwik.useStore([]);
  const selectedOptionSig = qwik.useSignal(props.defaultValue ? props.defaultValue : "");
  const isTriggerExpandedSig = qwik.useSignal(false);
  const inputRefSig = qwik.useSignal();
  const triggerRefSig = qwik.useSignal();
  const listBoxRefSig = qwik.useSignal();
  const rootRef = qwik.useSignal();
  const labelRef = qwik.useSignal();
  const inputValueSig = qwik.useSignal(props.defaultValue ? props.defaultValue : "");
  const listBoxId = qwik.useId();
  const inputId = qwik.useId();
  const triggerId = qwik.useId();
  const activeOptionId = qwik.useSignal(null);
  const isInputFocusedSig = qwik.useSignal(false);
  const focusInput$ = /* @__PURE__ */ qwik.inlinedQrl((inputId2) => {
    const [rootRef2] = qwik.useLexicalScope();
    rootRef2.value?.querySelector(`[data-autocomplete-input-id="${inputId2}"]`)?.focus();
  }, "AutocompleteRoot_component_focusInput_ympUatoO4zs", [
    rootRef
  ]);
  const dataHolder = qwik.useSignal([]);
  console.log(dataHolder.value);
  const contextService = {
    dataHolder,
    optionsStore,
    filteredOptionsStore,
    selectedOptionSig,
    isTriggerExpandedSig,
    inputRefSig,
    triggerRefSig,
    listBoxRefSig,
    labelRef,
    inputValueSig,
    listBoxId,
    inputId,
    triggerId,
    activeOptionId,
    focusInput$,
    isInputFocusedSig
  };
  qwik.useContextProvider(autocompleteContextId, contextService);
  const updatePosition = /* @__PURE__ */ qwik.inlinedQrl((referenceEl, floatingEl) => {
    dom.computePosition(referenceEl, floatingEl, {
      placement: "bottom",
      middleware: [
        dom.flip()
      ]
    }).then(({ x, y }) => {
      Object.assign(floatingEl.style, {
        left: `${x}px`,
        top: `${y}px`
      });
    });
  }, "AutocompleteRoot_component_updatePosition_C1fynSf09uw");
  qwik.useVisibleTaskQrl(/* @__PURE__ */ qwik.inlinedQrl(async function updatePositionTask({ track }) {
    const [inputRefSig2, isTriggerExpandedSig2, listBoxRefSig2, updatePosition2] = qwik.useLexicalScope();
    const inputRefValue = track(() => inputRefSig2.value);
    const listBox = track(() => listBoxRefSig2.value);
    const expanded = track(() => isTriggerExpandedSig2.value);
    if (!inputRefValue || !listBox)
      return;
    if (expanded === true)
      await updatePosition2(inputRefValue, listBox);
  }, "AutocompleteRoot_component_useVisibleTask_1_PDFlHYpiwiY", [
    inputRefSig,
    isTriggerExpandedSig,
    listBoxRefSig,
    updatePosition
  ]));
  qwik.useVisibleTaskQrl(/* @__PURE__ */ qwik.inlinedQrl(function filterOptionsTask({ track }) {
    const [contextService2, inputValueSig2, isInputFocusedSig2] = qwik.useLexicalScope();
    track(() => inputValueSig2.value);
    contextService2.filteredOptionsStore = contextService2.optionsStore.filter((option) => {
      const optionValue = option.value.getAttribute("optionValue");
      const defaultFilter = new RegExp(inputValueSig2.value, "i");
      if (isInputFocusedSig2.value) {
        if (optionValue === inputValueSig2.value)
          contextService2.isTriggerExpandedSig.value = false;
        else if (optionValue.match(defaultFilter))
          contextService2.isTriggerExpandedSig.value = true;
      } else
        contextService2.isTriggerExpandedSig.value = false;
      return optionValue.match(defaultFilter);
    });
    contextService2.optionsStore.map((option) => {
      if (!option.value.getAttribute("optionValue").match(new RegExp(inputValueSig2.value, "i")))
        option.value.style.display = "none";
      else
        option.value.style.display = "";
    });
  }, "AutocompleteRoot_component_useVisibleTask_2_N74uWMuubK0", [
    contextService,
    inputValueSig,
    isInputFocusedSig
  ]));
  qwik.useOnWindow("click", /* @__PURE__ */ qwik.inlinedQrl((e) => {
    const [contextService2] = qwik.useLexicalScope();
    const target = e.target;
    if (contextService2.isTriggerExpandedSig.value === true && !contextService2.listBoxRefSig.value?.contains(target) && !contextService2.triggerRefSig.value?.contains(target))
      contextService2.isTriggerExpandedSig.value = false;
  }, "AutocompleteRoot_component_useOnWindow_nlwR8BFcQiY", [
    contextService
  ]));
  return /* @__PURE__ */ qwik._jsxS("div", {
    ...props1,
    children: /* @__PURE__ */ qwik._jsxC(qwik.Slot, null, 3, "XJ_0"),
    onKeyDown$: [
      /* @__PURE__ */ qwik.inlinedQrl((e) => {
        const [contextService2] = qwik.useLexicalScope();
        if (e.key === "Escape") {
          contextService2.isTriggerExpandedSig.value = false;
          const inputElement = contextService2.inputRefSig.value?.firstElementChild;
          inputElement?.focus();
        }
      }, "AutocompleteRoot_component_div_onKeyDown_sTRTEBYUlQc", [
        contextService
      ]),
      props1.onKeyDown$
    ],
    ref: rootRef
  }, null, 0, "XJ_1");
}, "AutocompleteRoot_component_vXjTqQMRfek"));
exports.AutocompleteRoot = AutocompleteRoot;
