// @qwik-ui/headless/components/input/input.qwik.mjs
import { createContextId, componentQrl, inlinedQrl, useId, useSignal, useContextProvider, _jsxS, _jsxC, Slot, useContext } from "@builder.io/qwik";
import { InputPhone } from "../input-phone/input-phone.qwik.mjs";
const inputContext = createContextId("input-context-service");
const Root = /* @__PURE__ */ componentQrl(/* @__PURE__ */ inlinedQrl((props) => {
  const uniqueId = useId();
  const contextService = {
    ref: useSignal(),
    id: props.id || uniqueId
  };
  useContextProvider(inputContext, contextService);
  return /* @__PURE__ */ _jsxS("div", {
    ...props,
    children: /* @__PURE__ */ _jsxC(Slot, null, 3, "fW_0")
  }, null, 0, "fW_1");
}, "Root_component_iuaLFshWoJw"));
const Label = /* @__PURE__ */ componentQrl(/* @__PURE__ */ inlinedQrl((props) => {
  const context = useContext(inputContext);
  const id = props.id || context.id;
  return /* @__PURE__ */ _jsxS("label", {
    ...props,
    children: /* @__PURE__ */ _jsxC(Slot, null, 3, "fW_2"),
    for: id
  }, null, 0, "fW_3");
}, "Label_component_2yIkWCKvJ3c"));
const Hint = /* @__PURE__ */ componentQrl(/* @__PURE__ */ inlinedQrl((props) => {
  const context = useContext(inputContext);
  const id = props.id || context.id;
  return /* @__PURE__ */ _jsxS("div", {
    ...props,
    children: /* @__PURE__ */ _jsxC(Slot, null, 3, "fW_4"),
    id: `hint-${id}`
  }, null, 0, "fW_5");
}, "Hint_component_mu0m5sVo9mE"));
const Message = /* @__PURE__ */ componentQrl(/* @__PURE__ */ inlinedQrl((props) => {
  const context = useContext(inputContext);
  const id = props.id || context.id;
  return /* @__PURE__ */ _jsxS("div", {
    id: `message-${id}`,
    ...props,
    children: /* @__PURE__ */ _jsxC(Slot, null, 3, "fW_6")
  }, {
    "aria-live": "polite"
  }, 0, "fW_7");
}, "Message_component_cRA10a0JQ9k"));
const Phone = /* @__PURE__ */ componentQrl(/* @__PURE__ */ inlinedQrl((props) => {
  const context = useContext(inputContext);
  const id = props.id || context.id;
  return /* @__PURE__ */ _jsxC(InputPhone, {
    ...props,
    "aria-describedby": `hint-${id} message-${id}`,
    id
  }, 0, "fW_8");
}, "Phone_component_u4aRRXrWctM"));
export {
  Hint,
  Label,
  Message,
  Phone,
  Root,
  inputContext as _auto_inputContext
};
