// @qwik-ui/headless/components/input/input.qwik.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const qwik = require("@builder.io/qwik");
const inputPhone = require("../input-phone/input-phone.qwik.cjs");
const inputContext = qwik.createContextId("input-context-service");
const Root = /* @__PURE__ */ qwik.componentQrl(/* @__PURE__ */ qwik.inlinedQrl((props) => {
  const uniqueId = qwik.useId();
  const contextService = {
    ref: qwik.useSignal(),
    id: props.id || uniqueId
  };
  qwik.useContextProvider(inputContext, contextService);
  return /* @__PURE__ */ qwik._jsxS("div", {
    ...props,
    children: /* @__PURE__ */ qwik._jsxC(qwik.Slot, null, 3, "fW_0")
  }, null, 0, "fW_1");
}, "Root_component_iuaLFshWoJw"));
const Label = /* @__PURE__ */ qwik.componentQrl(/* @__PURE__ */ qwik.inlinedQrl((props) => {
  const context = qwik.useContext(inputContext);
  const id = props.id || context.id;
  return /* @__PURE__ */ qwik._jsxS("label", {
    ...props,
    children: /* @__PURE__ */ qwik._jsxC(qwik.Slot, null, 3, "fW_2"),
    for: id
  }, null, 0, "fW_3");
}, "Label_component_2yIkWCKvJ3c"));
const Hint = /* @__PURE__ */ qwik.componentQrl(/* @__PURE__ */ qwik.inlinedQrl((props) => {
  const context = qwik.useContext(inputContext);
  const id = props.id || context.id;
  return /* @__PURE__ */ qwik._jsxS("div", {
    ...props,
    children: /* @__PURE__ */ qwik._jsxC(qwik.Slot, null, 3, "fW_4"),
    id: `hint-${id}`
  }, null, 0, "fW_5");
}, "Hint_component_mu0m5sVo9mE"));
const Message = /* @__PURE__ */ qwik.componentQrl(/* @__PURE__ */ qwik.inlinedQrl((props) => {
  const context = qwik.useContext(inputContext);
  const id = props.id || context.id;
  return /* @__PURE__ */ qwik._jsxS("div", {
    id: `message-${id}`,
    ...props,
    children: /* @__PURE__ */ qwik._jsxC(qwik.Slot, null, 3, "fW_6")
  }, {
    "aria-live": "polite"
  }, 0, "fW_7");
}, "Message_component_cRA10a0JQ9k"));
const Phone = /* @__PURE__ */ qwik.componentQrl(/* @__PURE__ */ qwik.inlinedQrl((props) => {
  const context = qwik.useContext(inputContext);
  const id = props.id || context.id;
  return /* @__PURE__ */ qwik._jsxC(inputPhone.InputPhone, {
    ...props,
    "aria-describedby": `hint-${id} message-${id}`,
    id
  }, 0, "fW_8");
}, "Phone_component_u4aRRXrWctM"));
exports.Hint = Hint;
exports.Label = Label;
exports.Message = Message;
exports.Phone = Phone;
exports.Root = Root;
exports._auto_inputContext = inputContext;
