// @qwik-ui/headless/components/carousel/carousel.qwik.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const qwik = require("@builder.io/qwik");
const utils = require("@qwik-ui/utils");
const buttons = require("./styles/buttons.css.qwik.cjs");
const control = require("./styles/control.css.qwik.cjs");
const item = require("./styles/item.css.qwik.cjs");
const items = require("./styles/items.css.qwik.cjs");
const useCarousel = require("./use/use-carousel.qwik.cjs");
const useCarouselProvider = (state) => {
  qwik.useContextProvider(carouselContext, state);
};
const carouselContext = qwik.createContextId("carousel-root");
const Root = /* @__PURE__ */ qwik.componentQrl(/* @__PURE__ */ qwik.inlinedQrl((props) => {
  const props1 = qwik._restProps(props, [
    "use"
  ]);
  const carouselState = useCarousel.useCarousel();
  const provider = props.use || carouselState;
  useCarouselProvider(provider);
  return /* @__PURE__ */ qwik._jsxS("div", {
    id: qwik._wrapSignal(provider, "id"),
    ref: provider.ref,
    ...props1,
    children: /* @__PURE__ */ qwik._jsxC(qwik.Slot, null, 3, "d4_0")
  }, {
    role: "presentation"
  }, 0, "d4_1");
}, "Root_component_NRAWrRuxVfc"));
const ButtonNext = /* @__PURE__ */ qwik.componentQrl(/* @__PURE__ */ qwik.inlinedQrl((props) => {
  const props1 = qwik._restProps(props, [
    "onClick$",
    "label"
  ]);
  qwik.useStylesScopedQrl(/* @__PURE__ */ qwik.inlinedQrl(buttons, "ButtonNext_component_useStylesScoped_vMMW4PxXQ8w"));
  const context = qwik.useContext(carouselContext);
  return /* @__PURE__ */ qwik._jsxS("button", {
    ...props1,
    children: /* @__PURE__ */ qwik._jsxC(qwik.Slot, null, 3, "d4_2"),
    onClick$: [
      /* @__PURE__ */ qwik.inlinedQrl(() => {
        const [context2] = qwik.useLexicalScope();
        return context2.items.next();
      }, "ButtonNext_component_button_onClick_Qc9xlwbsj0Q", [
        context
      ]),
      props.onClick$
    ]
  }, {
    "aria-label": qwik._fnSignal((p0) => p0.label ?? "Go to the next item", [
      props
    ], 'p0.label??"Go to the next item"'),
    disabled: qwik._fnSignal((p0) => !p0.loop ? p0.items.active.isLast.value : false, [
      context
    ], "!p0.loop?p0.items.active.isLast.value:false")
  }, 0, "d4_3");
}, "ButtonNext_component_mi0CimL0jFI"));
const ButtonPrevious = /* @__PURE__ */ qwik.componentQrl(/* @__PURE__ */ qwik.inlinedQrl((props) => {
  const props1 = qwik._restProps(props, [
    "onClick$",
    "label"
  ]);
  qwik.useStylesScopedQrl(/* @__PURE__ */ qwik.inlinedQrl(buttons, "ButtonPrevious_component_useStylesScoped_fOvx4pn1zRQ"));
  const context = qwik.useContext(carouselContext);
  return /* @__PURE__ */ qwik._jsxS("button", {
    ...props1,
    children: /* @__PURE__ */ qwik._jsxC(qwik.Slot, null, 3, "d4_4"),
    onClick$: [
      /* @__PURE__ */ qwik.inlinedQrl(() => {
        const [context2] = qwik.useLexicalScope();
        return context2.items.previous();
      }, "ButtonPrevious_component_button_onClick_m4xC2sNSg0k", [
        context
      ]),
      props.onClick$
    ]
  }, {
    "aria-label": qwik._fnSignal((p0) => p0.label ?? "Go to the previous item", [
      props
    ], 'p0.label??"Go to the previous item"'),
    disabled: qwik._fnSignal((p0) => !p0.loop ? p0.items.active.isFirst.value : false, [
      context
    ], "!p0.loop?p0.items.active.isFirst.value:false")
  }, 0, "d4_5");
}, "ButtonPrevious_component_f2CDkWxKAsM"));
const Items = /* @__PURE__ */ qwik.componentQrl(/* @__PURE__ */ qwik.inlinedQrl((props) => {
  qwik.useStylesScopedQrl(/* @__PURE__ */ qwik.inlinedQrl(items, "Items_component_useStylesScoped_GpZR1tw0n0Q"));
  return /* @__PURE__ */ qwik._jsxQ("div", null, {
    "attr-data-qui": "carousel-wapper"
  }, /* @__PURE__ */ qwik._jsxS("ul", {
    ...props,
    children: /* @__PURE__ */ qwik._jsxC(qwik.Slot, null, 3, "d4_6")
  }, {
    "attr-data-qui": "carousel"
  }, 0, null), 1, "d4_7");
}, "Items_component_IQtmkwvbnI0"));
const Item = /* @__PURE__ */ qwik.componentQrl(/* @__PURE__ */ qwik.inlinedQrl((props) => {
  const props1 = qwik._restProps(props, [
    "index",
    "label"
  ]);
  qwik.useStylesScopedQrl(/* @__PURE__ */ qwik.inlinedQrl(item, "Item_component_useStylesScoped_lzrIZT9GSBU"));
  const context = qwik.useContext(carouselContext);
  return /* @__PURE__ */ qwik._jsxS("li", {
    ...props1,
    children: [
      /* @__PURE__ */ qwik._jsxQ("input", null, {
        "aria-label": qwik._fnSignal((p0) => p0.label, [
          props
        ], "p0.label"),
        checked: qwik._fnSignal((p0, p1) => p0.items.active.current.value === p1.index, [
          context,
          props
        ], "p0.items.active.current.value===p1.index"),
        name: qwik._fnSignal((p0) => `item-${p0.id}`, [
          context
        ], "`item-${p0.id}`"),
        onChange$: /* @__PURE__ */ qwik.inlinedQrl(() => {
          const [context2, props2] = qwik.useLexicalScope();
          return context2.items.scrollAt(props2.index);
        }, "Item_component_li_input_onChange_s56tIypzlRA", [
          context,
          props
        ]),
        type: "radio"
      }, null, 3, null),
      /* @__PURE__ */ qwik._jsxC(qwik.Slot, null, 3, "d4_8")
    ]
  }, {
    "aria-current": qwik._fnSignal((p0, p1) => p0.items.active.current.value === p1.index, [
      context,
      props
    ], "p0.items.active.current.value===p1.index")
  }, 0, "d4_9");
}, "Item_component_OYfNf5CqpyM"));
const controlContext = qwik.createContextId("carousel-control-root");
const Controls = /* @__PURE__ */ qwik.componentQrl(/* @__PURE__ */ qwik.inlinedQrl((props) => {
  qwik._jsxBranch();
  const uniqueId = qwik.useId();
  const controlService = {
    id: props.id || uniqueId
  };
  const context = qwik.useContext(carouselContext);
  qwik.useContextProvider(controlContext, controlService);
  return /* @__PURE__ */ qwik._jsxS("nav", {
    ...props,
    children: context.pages.ranges.value && /* @__PURE__ */ qwik._jsxC(qwik.Slot, null, 3, "d4_10")
  }, null, 0, "d4_11");
}, "Controls_component_tsn2aW4QuKA"));
const Control = /* @__PURE__ */ qwik.componentQrl(/* @__PURE__ */ qwik.inlinedQrl((props) => {
  const props1 = qwik._restProps(props, [
    "index",
    "onClick$",
    "label"
  ]);
  qwik.useStylesScopedQrl(/* @__PURE__ */ qwik.inlinedQrl(control, "Control_component_useStylesScoped_L2t1ZYaMmtQ"));
  const ordinal = utils.useOrdinal();
  const context = qwik.useContext(carouselContext);
  const context1 = qwik.useContext(controlContext);
  return /* @__PURE__ */ qwik._jsxS("div", {
    ...props1,
    children: [
      /* @__PURE__ */ qwik._jsxQ("input", null, {
        "aria-label": qwik._fnSignal((p0, p1) => p1.label || `Go to the ${p0?.(p1.index + 1)} item`, [
          ordinal,
          props
        ], "p1.label||`Go to the ${p0?.(p1.index+1)} item`"),
        checked: qwik._fnSignal((p0, p1) => p0.items.active.current.value === p1.index, [
          context,
          props
        ], "p0.items.active.current.value===p1.index"),
        name: qwik._fnSignal((p0) => `control-${p0.id}`, [
          context1
        ], "`control-${p0.id}`"),
        onChange$: /* @__PURE__ */ qwik.inlinedQrl(() => {
          const [context2, props2] = qwik.useLexicalScope();
          return context2.items.scrollAt(props2.index);
        }, "Control_component_div_input_onChange_KMuD8ad6BIY", [
          context,
          props
        ]),
        type: "radio"
      }, null, 3, null),
      /* @__PURE__ */ qwik._jsxC(qwik.Slot, null, 3, "d4_12")
    ],
    onClick$: [
      /* @__PURE__ */ qwik.inlinedQrl(() => {
        const [context2, props2] = qwik.useLexicalScope();
        return context2.items.scrollAt(props2.index);
      }, "Control_component_div_onClick_046SLdYoam0", [
        context,
        props
      ]),
      props.onClick$
    ]
  }, {
    "aria-current": qwik._fnSignal((p0, p1) => p0.items.active.current.value === p1.index, [
      context,
      props
    ], "p0.items.active.current.value===p1.index")
  }, 0, "d4_13");
}, "Control_component_KjEWhtG7Dp4"));
const IconPrevious = () => /* @__PURE__ */ qwik._jsxQ("svg", null, {
  fill: "none",
  height: "24",
  stroke: "currentColor",
  "stroke-linecap": "round",
  "stroke-linejoin": "round",
  "stroke-width": "2",
  viewBox: "0 0 24 24",
  width: "24",
  xmlns: "http://www.w3.org/2000/svg"
}, /* @__PURE__ */ qwik._jsxQ("polyline", null, {
  points: "15 18 9 12 15 6"
}, null, 3, null), 3, "d4_14");
const IconNext = () => /* @__PURE__ */ qwik._jsxQ("svg", null, {
  fill: "none",
  height: "24",
  stroke: "currentColor",
  "stroke-linecap": "round",
  "stroke-linejoin": "round",
  "stroke-width": "2",
  viewBox: "0 0 24 24",
  width: "24",
  xmlns: "http://www.w3.org/2000/svg"
}, /* @__PURE__ */ qwik._jsxQ("polyline", null, {
  points: "9 18 15 12 9 6"
}, null, 3, null), 3, "d4_15");
const IconChevronLeft = () => /* @__PURE__ */ qwik._jsxQ("svg", null, {
  class: "lucide lucide-chevrons-left",
  fill: "none",
  height: "24",
  stroke: "currentColor",
  "stroke-linecap": "round",
  "stroke-linejoin": "round",
  "stroke-width": "2",
  viewBox: "0 0 24 24",
  width: "24",
  xmlns: "http://www.w3.org/2000/svg"
}, [
  /* @__PURE__ */ qwik._jsxQ("polyline", null, {
    points: "11 17 6 12 11 7"
  }, null, 3, null),
  /* @__PURE__ */ qwik._jsxQ("polyline", null, {
    points: "18 17 13 12 18 7"
  }, null, 3, null)
], 3, "d4_16");
const IconChevronRight = () => /* @__PURE__ */ qwik._jsxQ("svg", null, {
  class: "lucide lucide-chevrons-right",
  fill: "none",
  height: "24",
  stroke: "currentColor",
  "stroke-linecap": "round",
  "stroke-linejoin": "round",
  "stroke-width": "2",
  viewBox: "0 0 24 24",
  width: "24",
  xmlns: "http://www.w3.org/2000/svg"
}, [
  /* @__PURE__ */ qwik._jsxQ("polyline", null, {
    points: "13 17 18 12 13 7"
  }, null, 3, null),
  /* @__PURE__ */ qwik._jsxQ("polyline", null, {
    points: "6 17 11 12 6 7"
  }, null, 3, null)
], 3, "d4_17");
exports.ButtonNext = ButtonNext;
exports.ButtonPrevious = ButtonPrevious;
exports.Control = Control;
exports.Controls = Controls;
exports.IconChevronLeft = IconChevronLeft;
exports.IconChevronRight = IconChevronRight;
exports.IconNext = IconNext;
exports.IconPrevious = IconPrevious;
exports.Item = Item;
exports.Items = Items;
exports.Root = Root;
exports.carouselContext = carouselContext;
exports.controlContext = controlContext;
exports.useCarouselProvider = useCarouselProvider;
