// @qwik-ui/headless/components/carousel/carousel.qwik.mjs
import { createContextId, useContextProvider, componentQrl, inlinedQrl, _restProps, _jsxS, _wrapSignal, _jsxC, Slot, useStylesScopedQrl, useContext, useLexicalScope, _fnSignal, _jsxQ, _jsxBranch, useId } from "@builder.io/qwik";
import { useOrdinal } from "@qwik-ui/utils";
import stylesButtons from "./styles/buttons.css.qwik.mjs";
import stylesControl from "./styles/control.css.qwik.mjs";
import stylesItem from "./styles/item.css.qwik.mjs";
import stylesItems from "./styles/items.css.qwik.mjs";
import { useCarousel } from "./use/use-carousel.qwik.mjs";
const useCarouselProvider = (state) => {
  useContextProvider(carouselContext, state);
};
const carouselContext = createContextId("carousel-root");
const Root = /* @__PURE__ */ componentQrl(/* @__PURE__ */ inlinedQrl((props) => {
  const props1 = _restProps(props, [
    "use"
  ]);
  const carouselState = useCarousel();
  const provider = props.use || carouselState;
  useCarouselProvider(provider);
  return /* @__PURE__ */ _jsxS("div", {
    id: _wrapSignal(provider, "id"),
    ref: provider.ref,
    ...props1,
    children: /* @__PURE__ */ _jsxC(Slot, null, 3, "d4_0")
  }, {
    role: "presentation"
  }, 0, "d4_1");
}, "Root_component_NRAWrRuxVfc"));
const ButtonNext = /* @__PURE__ */ componentQrl(/* @__PURE__ */ inlinedQrl((props) => {
  const props1 = _restProps(props, [
    "onClick$",
    "label"
  ]);
  useStylesScopedQrl(/* @__PURE__ */ inlinedQrl(stylesButtons, "ButtonNext_component_useStylesScoped_vMMW4PxXQ8w"));
  const context = useContext(carouselContext);
  return /* @__PURE__ */ _jsxS("button", {
    ...props1,
    children: /* @__PURE__ */ _jsxC(Slot, null, 3, "d4_2"),
    onClick$: [
      /* @__PURE__ */ inlinedQrl(() => {
        const [context2] = useLexicalScope();
        return context2.items.next();
      }, "ButtonNext_component_button_onClick_Qc9xlwbsj0Q", [
        context
      ]),
      props.onClick$
    ]
  }, {
    "aria-label": _fnSignal((p0) => p0.label ?? "Go to the next item", [
      props
    ], 'p0.label??"Go to the next item"'),
    disabled: _fnSignal((p0) => !p0.loop ? p0.items.active.isLast.value : false, [
      context
    ], "!p0.loop?p0.items.active.isLast.value:false")
  }, 0, "d4_3");
}, "ButtonNext_component_mi0CimL0jFI"));
const ButtonPrevious = /* @__PURE__ */ componentQrl(/* @__PURE__ */ inlinedQrl((props) => {
  const props1 = _restProps(props, [
    "onClick$",
    "label"
  ]);
  useStylesScopedQrl(/* @__PURE__ */ inlinedQrl(stylesButtons, "ButtonPrevious_component_useStylesScoped_fOvx4pn1zRQ"));
  const context = useContext(carouselContext);
  return /* @__PURE__ */ _jsxS("button", {
    ...props1,
    children: /* @__PURE__ */ _jsxC(Slot, null, 3, "d4_4"),
    onClick$: [
      /* @__PURE__ */ inlinedQrl(() => {
        const [context2] = useLexicalScope();
        return context2.items.previous();
      }, "ButtonPrevious_component_button_onClick_m4xC2sNSg0k", [
        context
      ]),
      props.onClick$
    ]
  }, {
    "aria-label": _fnSignal((p0) => p0.label ?? "Go to the previous item", [
      props
    ], 'p0.label??"Go to the previous item"'),
    disabled: _fnSignal((p0) => !p0.loop ? p0.items.active.isFirst.value : false, [
      context
    ], "!p0.loop?p0.items.active.isFirst.value:false")
  }, 0, "d4_5");
}, "ButtonPrevious_component_f2CDkWxKAsM"));
const Items = /* @__PURE__ */ componentQrl(/* @__PURE__ */ inlinedQrl((props) => {
  useStylesScopedQrl(/* @__PURE__ */ inlinedQrl(stylesItems, "Items_component_useStylesScoped_GpZR1tw0n0Q"));
  return /* @__PURE__ */ _jsxQ("div", null, {
    "attr-data-qui": "carousel-wapper"
  }, /* @__PURE__ */ _jsxS("ul", {
    ...props,
    children: /* @__PURE__ */ _jsxC(Slot, null, 3, "d4_6")
  }, {
    "attr-data-qui": "carousel"
  }, 0, null), 1, "d4_7");
}, "Items_component_IQtmkwvbnI0"));
const Item = /* @__PURE__ */ componentQrl(/* @__PURE__ */ inlinedQrl((props) => {
  const props1 = _restProps(props, [
    "index",
    "label"
  ]);
  useStylesScopedQrl(/* @__PURE__ */ inlinedQrl(stylesItem, "Item_component_useStylesScoped_lzrIZT9GSBU"));
  const context = useContext(carouselContext);
  return /* @__PURE__ */ _jsxS("li", {
    ...props1,
    children: [
      /* @__PURE__ */ _jsxQ("input", null, {
        "aria-label": _fnSignal((p0) => p0.label, [
          props
        ], "p0.label"),
        checked: _fnSignal((p0, p1) => p0.items.active.current.value === p1.index, [
          context,
          props
        ], "p0.items.active.current.value===p1.index"),
        name: _fnSignal((p0) => `item-${p0.id}`, [
          context
        ], "`item-${p0.id}`"),
        onChange$: /* @__PURE__ */ inlinedQrl(() => {
          const [context2, props2] = useLexicalScope();
          return context2.items.scrollAt(props2.index);
        }, "Item_component_li_input_onChange_s56tIypzlRA", [
          context,
          props
        ]),
        type: "radio"
      }, null, 3, null),
      /* @__PURE__ */ _jsxC(Slot, null, 3, "d4_8")
    ]
  }, {
    "aria-current": _fnSignal((p0, p1) => p0.items.active.current.value === p1.index, [
      context,
      props
    ], "p0.items.active.current.value===p1.index")
  }, 0, "d4_9");
}, "Item_component_OYfNf5CqpyM"));
const controlContext = createContextId("carousel-control-root");
const Controls = /* @__PURE__ */ componentQrl(/* @__PURE__ */ inlinedQrl((props) => {
  _jsxBranch();
  const uniqueId = useId();
  const controlService = {
    id: props.id || uniqueId
  };
  const context = useContext(carouselContext);
  useContextProvider(controlContext, controlService);
  return /* @__PURE__ */ _jsxS("nav", {
    ...props,
    children: context.pages.ranges.value && /* @__PURE__ */ _jsxC(Slot, null, 3, "d4_10")
  }, null, 0, "d4_11");
}, "Controls_component_tsn2aW4QuKA"));
const Control = /* @__PURE__ */ componentQrl(/* @__PURE__ */ inlinedQrl((props) => {
  const props1 = _restProps(props, [
    "index",
    "onClick$",
    "label"
  ]);
  useStylesScopedQrl(/* @__PURE__ */ inlinedQrl(stylesControl, "Control_component_useStylesScoped_L2t1ZYaMmtQ"));
  const ordinal = useOrdinal();
  const context = useContext(carouselContext);
  const context1 = useContext(controlContext);
  return /* @__PURE__ */ _jsxS("div", {
    ...props1,
    children: [
      /* @__PURE__ */ _jsxQ("input", null, {
        "aria-label": _fnSignal((p0, p1) => p1.label || `Go to the ${p0?.(p1.index + 1)} item`, [
          ordinal,
          props
        ], "p1.label||`Go to the ${p0?.(p1.index+1)} item`"),
        checked: _fnSignal((p0, p1) => p0.items.active.current.value === p1.index, [
          context,
          props
        ], "p0.items.active.current.value===p1.index"),
        name: _fnSignal((p0) => `control-${p0.id}`, [
          context1
        ], "`control-${p0.id}`"),
        onChange$: /* @__PURE__ */ inlinedQrl(() => {
          const [context2, props2] = useLexicalScope();
          return context2.items.scrollAt(props2.index);
        }, "Control_component_div_input_onChange_KMuD8ad6BIY", [
          context,
          props
        ]),
        type: "radio"
      }, null, 3, null),
      /* @__PURE__ */ _jsxC(Slot, null, 3, "d4_12")
    ],
    onClick$: [
      /* @__PURE__ */ inlinedQrl(() => {
        const [context2, props2] = useLexicalScope();
        return context2.items.scrollAt(props2.index);
      }, "Control_component_div_onClick_046SLdYoam0", [
        context,
        props
      ]),
      props.onClick$
    ]
  }, {
    "aria-current": _fnSignal((p0, p1) => p0.items.active.current.value === p1.index, [
      context,
      props
    ], "p0.items.active.current.value===p1.index")
  }, 0, "d4_13");
}, "Control_component_KjEWhtG7Dp4"));
const IconPrevious = () => /* @__PURE__ */ _jsxQ("svg", null, {
  fill: "none",
  height: "24",
  stroke: "currentColor",
  "stroke-linecap": "round",
  "stroke-linejoin": "round",
  "stroke-width": "2",
  viewBox: "0 0 24 24",
  width: "24",
  xmlns: "http://www.w3.org/2000/svg"
}, /* @__PURE__ */ _jsxQ("polyline", null, {
  points: "15 18 9 12 15 6"
}, null, 3, null), 3, "d4_14");
const IconNext = () => /* @__PURE__ */ _jsxQ("svg", null, {
  fill: "none",
  height: "24",
  stroke: "currentColor",
  "stroke-linecap": "round",
  "stroke-linejoin": "round",
  "stroke-width": "2",
  viewBox: "0 0 24 24",
  width: "24",
  xmlns: "http://www.w3.org/2000/svg"
}, /* @__PURE__ */ _jsxQ("polyline", null, {
  points: "9 18 15 12 9 6"
}, null, 3, null), 3, "d4_15");
const IconChevronLeft = () => /* @__PURE__ */ _jsxQ("svg", null, {
  class: "lucide lucide-chevrons-left",
  fill: "none",
  height: "24",
  stroke: "currentColor",
  "stroke-linecap": "round",
  "stroke-linejoin": "round",
  "stroke-width": "2",
  viewBox: "0 0 24 24",
  width: "24",
  xmlns: "http://www.w3.org/2000/svg"
}, [
  /* @__PURE__ */ _jsxQ("polyline", null, {
    points: "11 17 6 12 11 7"
  }, null, 3, null),
  /* @__PURE__ */ _jsxQ("polyline", null, {
    points: "18 17 13 12 18 7"
  }, null, 3, null)
], 3, "d4_16");
const IconChevronRight = () => /* @__PURE__ */ _jsxQ("svg", null, {
  class: "lucide lucide-chevrons-right",
  fill: "none",
  height: "24",
  stroke: "currentColor",
  "stroke-linecap": "round",
  "stroke-linejoin": "round",
  "stroke-width": "2",
  viewBox: "0 0 24 24",
  width: "24",
  xmlns: "http://www.w3.org/2000/svg"
}, [
  /* @__PURE__ */ _jsxQ("polyline", null, {
    points: "13 17 18 12 13 7"
  }, null, 3, null),
  /* @__PURE__ */ _jsxQ("polyline", null, {
    points: "6 17 11 12 6 7"
  }, null, 3, null)
], 3, "d4_17");
export {
  ButtonNext,
  ButtonPrevious,
  Control,
  Controls,
  IconChevronLeft,
  IconChevronRight,
  IconNext,
  IconPrevious,
  Item,
  Items,
  Root,
  carouselContext,
  controlContext,
  useCarouselProvider
};
