// @qwik-ui/headless/components/combobox/combobox-listbox.qwik.mjs
import { componentQrl, inlinedQrl, _restProps, useContext, useVisibleTaskQrl, useLexicalScope, _jsxS, _fnSignal } from "@builder.io/qwik";
import { autoUpdate, offset, hide, computePosition, flip, shift, autoPlacement } from "@floating-ui/dom";
import ComboboxContextId from "./combobox-context-id.qwik.mjs";
const ComboboxListbox = /* @__PURE__ */ componentQrl(/* @__PURE__ */ inlinedQrl((props) => {
  const props1 = _restProps(props, [
    "optionRenderer$",
    "gutter",
    "flip",
    "placement",
    "shift",
    "hide",
    "autoPlacement",
    "ancestorScroll",
    "ancestorResize",
    "elementResize",
    "animationFrame",
    "transform"
  ]);
  const context = useContext(ComboboxContextId);
  const listboxId = `${context.localId}-listbox`;
  useVisibleTaskQrl(/* @__PURE__ */ inlinedQrl(function setFloatingUIConfig({ cleanup }) {
    const [context2, props2] = useLexicalScope();
    function updatePosition() {
      const middleware = [
        offset(props2.gutter),
        hide({
          strategy: props2.hide
        })
      ];
      const middlewareFunctions = [
        flip,
        shift,
        autoPlacement
      ];
      const middlewareProps = [
        props2.flip ?? true,
        props2.shift,
        props2.autoPlacement ?? false
      ];
      middlewareFunctions.forEach((func, index) => {
        const isMiddlewareEnabled = middlewareProps[index];
        if (isMiddlewareEnabled)
          middleware.push(func());
      });
      computePosition(context2.inputRef.value, context2.listboxRef.value, {
        placement: props2.placement ?? "bottom",
        middleware
      }).then((resolvedData) => {
        const { x, y } = resolvedData;
        if (context2.listboxRef.value)
          Object.assign(context2.listboxRef.value.style, {
            left: `${x}px`,
            top: `${y}px`,
            transform: props2.transform
          });
      });
    }
    if (context2.inputRef.value && context2.listboxRef.value) {
      updatePosition();
      const cleanupFunc = autoUpdate(context2.inputRef.value, context2.listboxRef.value, updatePosition, {
        ancestorScroll: props2.ancestorScroll ?? true,
        ancestorResize: props2.ancestorResize ?? true,
        elementResize: props2.elementResize ?? true,
        animationFrame: props2.animationFrame ?? false
      });
      cleanup(() => {
        cleanupFunc();
      });
    }
  }, "ComboboxListbox_component_useVisibleTask_CC7IMsQgQvI", [
    context,
    props
  ]));
  return /* @__PURE__ */ _jsxS("ul", {
    ...props1,
    children: context.filteredOptionsSig.value.map((resolved, filteredIndex) => props.optionRenderer$?.(resolved, filteredIndex)),
    id: listboxId,
    ref: context.listboxRef,
    style: {
      ...props1.style,
      position: "absolute"
    }
  }, {
    "aria-label": _fnSignal((p0) => p0.labelRef.value ? p0.labelRef.value?.innerText : "Suggestions", [
      context
    ], 'p0.labelRef.value?p0.labelRef.value?.innerText:"Suggestions"'),
    role: "listbox"
  }, 0, "Jv_0");
}, "ComboboxListbox_component_oCbY22ovk0I"));
export {
  ComboboxListbox
};
