// @qwik-ui/headless/components/combobox/combobox-input.qwik.mjs
import { componentQrl, inlinedQrl, _restProps, useContext, useSignal, useTaskQrl, useLexicalScope, _jsxS, _fnSignal } from "@builder.io/qwik";
import ComboboxContextId from "./combobox-context-id.qwik.mjs";
import { getActiveDescendant, getNextEnabledOptionIndex, getPrevEnabledOptionIndex } from "./utils.qwik.mjs";
const ComboboxInput = /* @__PURE__ */ componentQrl(/* @__PURE__ */ inlinedQrl((props) => {
  const props1 = _restProps(props, [
    "disableOnBlur"
  ]);
  const context = useContext(ComboboxContextId);
  const inputId = props1.id || `${context.localId}-input`;
  const listboxId = `${context.localId}-listbox`;
  const isDefaultLabelNeededSig = useSignal(true);
  const onInputBehavior$ = /* @__PURE__ */ inlinedQrl((e) => {
    const [context2] = useLexicalScope();
    context2.isListboxOpenSig.value = true;
    context2.highlightedIndexSig.value = -1;
    const inputElement = e.target;
    context2.inputValueSig.value = inputElement.value;
  }, "ComboboxInput_component_onInputBehavior_H0TYIoxEvN0", [
    context
  ]);
  const onKeydownBehavior$ = /* @__PURE__ */ inlinedQrl((e) => {
    const [context2] = useLexicalScope();
    if (e.key === "ArrowDown") {
      if (context2.isListboxOpenSig.value) {
        const nextEnabledOptionIndex = getNextEnabledOptionIndex(context2.highlightedIndexSig.value, context2.filteredOptionsSig);
        context2.highlightedIndexSig.value = nextEnabledOptionIndex;
      } else if (context2.highlightedIndexSig.value === -1) {
        const firstEnabledOptionIndex = getNextEnabledOptionIndex(-1, context2.filteredOptionsSig);
        context2.highlightedIndexSig.value = firstEnabledOptionIndex;
      }
      context2.isListboxOpenSig.value = true;
    }
    if (e.key === "ArrowUp") {
      const prevEnabledOptionIndex = getPrevEnabledOptionIndex(context2.highlightedIndexSig.value, context2.filteredOptionsSig);
      context2.highlightedIndexSig.value = prevEnabledOptionIndex;
    }
    if (e.key === "Enter") {
      context2.isListboxOpenSig.value = false;
      if (context2.filteredOptionsSig.value[context2.highlightedIndexSig.value].disabled)
        return;
      const inputElement = e.target;
      inputElement.value = context2.filteredOptionsSig.value[context2.highlightedIndexSig.value].label;
    }
    if (e.key === "Home") {
      const firstEnabledOptionIndex = getNextEnabledOptionIndex(-1, context2.filteredOptionsSig);
      context2.highlightedIndexSig.value = firstEnabledOptionIndex;
    }
    if (e.key === "End") {
      const lastEnabledOptionIndex = getPrevEnabledOptionIndex(context2.filteredOptionsSig.value.length, context2.filteredOptionsSig);
      context2.highlightedIndexSig.value = lastEnabledOptionIndex;
    }
    if (e.key === "Escape")
      context2.isListboxOpenSig.value = false;
  }, "ComboboxInput_component_onKeydownBehavior_VbW6nXYoVV8", [
    context
  ]);
  useTaskQrl(/* @__PURE__ */ inlinedQrl(function isLabelNeededTask() {
    const [context2, isDefaultLabelNeededSig2] = useLexicalScope();
    if (!context2.inputRef.value || !context2.defaultLabel)
      return;
    if (context2.inputRef.value.value === context2.defaultLabel)
      isDefaultLabelNeededSig2.value = false;
  }, "ComboboxInput_component_useTask_0fw3VrSiM5I", [
    context,
    isDefaultLabelNeededSig
  ]));
  useTaskQrl(/* @__PURE__ */ inlinedQrl(function highlightDefaultLabelTask() {
    const [context2] = useLexicalScope();
    const defaultIndex = context2.filteredOptionsSig.value.findIndex(({ label }) => label === context2.defaultLabel);
    if (defaultIndex !== -1)
      context2.highlightedIndexSig.value = defaultIndex;
  }, "ComboboxInput_component_useTask_1_to0vbTbWOHo", [
    context
  ]));
  return /* @__PURE__ */ _jsxS("input", {
    ...props1,
    "aria-activedescendant": context.isListboxOpenSig.value ? getActiveDescendant(context.highlightedIndexSig, context.filteredOptionsSig, context.localId) : "",
    "aria-controls": listboxId,
    id: inputId || props1.id,
    onBlur$: [
      /* @__PURE__ */ inlinedQrl(() => {
        const [context2, props2] = useLexicalScope();
        (props2.disableOnBlur ?? false) || (context2.isListboxOpenSig.value = false);
      }, "ComboboxInput_component_input_onBlur_zwsk0Nv2s9M", [
        context,
        props
      ]),
      props1.onBlur$
    ],
    onInput$: [
      onInputBehavior$,
      props1.onInput$
    ],
    onKeyDown$: [
      onKeydownBehavior$,
      props1.onKeyDown$
    ],
    ref: context.inputRef
  }, {
    "aria-autocomplete": "list",
    "aria-expanded": _fnSignal((p0) => p0.isListboxOpenSig.value, [
      context
    ], "p0.isListboxOpenSig.value"),
    "aria-haspopup": "listbox",
    role: "combobox",
    type: "text",
    value: _fnSignal((p0) => p0.inputValueSig.value, [
      context
    ], "p0.inputValueSig.value")
  }, 0, "XG_0");
}, "ComboboxInput_component_ZWi3j8rQMQU"));
export {
  ComboboxInput
};
