// @qwik-ui/headless/components/combobox/combobox-option.qwik.mjs
import { componentQrl, inlinedQrl, _restProps, useContext, useComputedQrl, useLexicalScope, useSignal, _jsxS, _jsxC, Slot, _fnSignal } from "@builder.io/qwik";
import ComboboxContextId from "./combobox-context-id.qwik.mjs";
const ComboboxOption = /* @__PURE__ */ componentQrl(/* @__PURE__ */ inlinedQrl(
  // remove non-li props from props
  (props) => {
    const props1 = _restProps(props, [
      "index",
      "resolved"
    ]);
    const context = useContext(ComboboxContextId);
    const optionId = `${context.localId}-${props.resolved.key}`;
    const isHighlightedSig = useComputedQrl(/* @__PURE__ */ inlinedQrl(
      // eslint-disable-next-line qwik/valid-lexical-scope
      () => {
        const [context2, props2] = useLexicalScope();
        return !props2.resolved.disabled && context2.highlightedIndexSig.value === props2.index;
      },
      "ComboboxOption_component_isHighlightedSig_useComputed_ew4kpEj9RH4",
      [
        context,
        props
      ]
    ));
    const onClickBehavior$ = /* @__PURE__ */ inlinedQrl(() => {
      const [context2, props2] = useLexicalScope();
      if (!context2.inputRef.value || props2.resolved.disabled)
        return;
      context2.inputRef.value.value = context2.filteredOptionsSig.value[props2.index]?.label, context2.isListboxOpenSig.value = false;
    }, "ComboboxOption_component_onClickBehavior_02F6OSkZ1Hw", [
      context,
      props
    ]);
    const optionRef = useSignal();
    return /* @__PURE__ */ _jsxS("li", {
      ...props1,
      children: /* @__PURE__ */ _jsxC(Slot, null, 3, "1r_0"),
      id: optionId,
      onClick$: [
        onClickBehavior$,
        props1.onClick$
      ],
      onMouseEnter$: [
        /* @__PURE__ */ inlinedQrl(() => {
          const [context2, props2] = useLexicalScope();
          return context2.highlightedIndexSig.value = props2.index;
        }, "ComboboxOption_component_li_onMouseEnter_OvCdpat00Yg", [
          context,
          props
        ]),
        props1.onMouseEnter$
      ],
      ref: optionRef
    }, {
      "aria-disabled": _fnSignal((p0) => p0.resolved.disabled, [
        props
      ], "p0.resolved.disabled"),
      "aria-selected": _fnSignal((p0) => p0.value, [
        isHighlightedSig
      ], "p0.value"),
      "data-disabled": _fnSignal((p0) => p0.resolved.disabled, [
        props
      ], "p0.resolved.disabled"),
      "preventdefault:mousedown": true,
      role: "option",
      tabIndex: 0
    }, 0, "1r_1");
  },
  "ComboboxOption_component_uqfuQGinDQI"
));
export {
  ComboboxOption
};
