// @qwik-ui/headless/components/combobox/combobox-option.qwik.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const qwik = require("@builder.io/qwik");
const comboboxContextId = require("./combobox-context-id.qwik.cjs");
const ComboboxOption = /* @__PURE__ */ qwik.componentQrl(/* @__PURE__ */ qwik.inlinedQrl(
  // remove non-li props from props
  (props) => {
    const props1 = qwik._restProps(props, [
      "index",
      "resolved"
    ]);
    const context = qwik.useContext(comboboxContextId);
    const optionId = `${context.localId}-${props.resolved.key}`;
    const isHighlightedSig = qwik.useComputedQrl(/* @__PURE__ */ qwik.inlinedQrl(
      // eslint-disable-next-line qwik/valid-lexical-scope
      () => {
        const [context2, props2] = qwik.useLexicalScope();
        return !props2.resolved.disabled && context2.highlightedIndexSig.value === props2.index;
      },
      "ComboboxOption_component_isHighlightedSig_useComputed_ew4kpEj9RH4",
      [
        context,
        props
      ]
    ));
    const onClickBehavior$ = /* @__PURE__ */ qwik.inlinedQrl(() => {
      const [context2, props2] = qwik.useLexicalScope();
      if (!context2.inputRef.value || props2.resolved.disabled)
        return;
      context2.inputRef.value.value = context2.filteredOptionsSig.value[props2.index]?.label, context2.isListboxOpenSig.value = false;
    }, "ComboboxOption_component_onClickBehavior_02F6OSkZ1Hw", [
      context,
      props
    ]);
    const optionRef = qwik.useSignal();
    return /* @__PURE__ */ qwik._jsxS("li", {
      ...props1,
      children: /* @__PURE__ */ qwik._jsxC(qwik.Slot, null, 3, "1r_0"),
      id: optionId,
      onClick$: [
        onClickBehavior$,
        props1.onClick$
      ],
      onMouseEnter$: [
        /* @__PURE__ */ qwik.inlinedQrl(() => {
          const [context2, props2] = qwik.useLexicalScope();
          return context2.highlightedIndexSig.value = props2.index;
        }, "ComboboxOption_component_li_onMouseEnter_OvCdpat00Yg", [
          context,
          props
        ]),
        props1.onMouseEnter$
      ],
      ref: optionRef
    }, {
      "aria-disabled": qwik._fnSignal((p0) => p0.resolved.disabled, [
        props
      ], "p0.resolved.disabled"),
      "aria-selected": qwik._fnSignal((p0) => p0.value, [
        isHighlightedSig
      ], "p0.value"),
      "data-disabled": qwik._fnSignal((p0) => p0.resolved.disabled, [
        props
      ], "p0.resolved.disabled"),
      "preventdefault:mousedown": true,
      role: "option",
      tabIndex: 0
    }, 0, "1r_1");
  },
  "ComboboxOption_component_uqfuQGinDQI"
));
exports.ComboboxOption = ComboboxOption;
