// @qwik-ui/headless/components/tooltip/tooltip.qwik.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const qwik = require("@builder.io/qwik");
const jsxRuntime = require("@builder.io/qwik/jsx-runtime");
const dom = require("@floating-ui/dom");
const tooltip = require("./tooltip.css.qwik.cjs");
const Tooltip = /* @__PURE__ */ qwik.componentQrl(/* @__PURE__ */ qwik.inlinedQrl((props) => {
  const props1 = qwik._restProps(props, [
    "offset",
    "content",
    "position",
    "durationMs"
  ]);
  qwik.useStylesScopedQrl(/* @__PURE__ */ qwik.inlinedQrl(tooltip, "Tooltip_component_useStylesScoped_3jJQGcomGNw"));
  const id = qwik.useId();
  const triggerAnchor = qwik.useSignal();
  const tooltipAnchor = qwik.useSignal();
  const stateSignal = qwik.useSignal("closing");
  const positionSignal = qwik.useSignal({
    x: 0,
    y: 0
  });
  const Wrapper = props1.inline ? "span" : "div";
  const lastActivatedTimestamp = qwik.useSignal(Date.now());
  const update = /* @__PURE__ */ qwik.inlinedQrl(async () => {
    const [lastActivatedTimestamp2, positionSignal2, props2, stateSignal2, tooltipAnchor2, triggerAnchor2] = qwik.useLexicalScope();
    const now = Date.now();
    const middleware = [
      dom.offset(props2.offset)
    ];
    const hasMouseEnterDebounced = now - lastActivatedTimestamp2.value >= 300;
    if (triggerAnchor2.value && tooltipAnchor2.value && hasMouseEnterDebounced) {
      const { x, y } = await dom.computePosition(triggerAnchor2.value, tooltipAnchor2.value, {
        placement: props2.position ?? "top",
        middleware
      });
      lastActivatedTimestamp2.value = now;
      positionSignal2.value = {
        x,
        y
      };
      stateSignal2.value = "positioned";
    }
  }, "Tooltip_component_update_b7EpqFj07nQ", [
    lastActivatedTimestamp,
    positionSignal,
    props,
    stateSignal,
    tooltipAnchor,
    triggerAnchor
  ]);
  const showTooltip = /* @__PURE__ */ qwik.inlinedQrl(() => {
    const [props2, stateSignal2, update2] = qwik.useLexicalScope();
    update2();
    setTimeout(() => stateSignal2.value = "positioned", props2.durationMs ?? 100);
  }, "Tooltip_component_showTooltip_f0VZ50EMnCA", [
    props,
    stateSignal,
    update
  ]);
  const hideTooltip = /* @__PURE__ */ qwik.inlinedQrl(() => {
    const [props2, stateSignal2] = qwik.useLexicalScope();
    setTimeout(() => stateSignal2.value = "closing", props2.durationMs ?? 100);
  }, "Tooltip_component_hideTooltip_o30x7aVxT7s", [
    props,
    stateSignal
  ]);
  qwik.useOnWindow("keyup", /* @__PURE__ */ qwik.inlinedQrl((e) => {
    const [hideTooltip2] = qwik.useLexicalScope();
    const key = e.key;
    if (key === "Escape")
      hideTooltip2();
  }, "Tooltip_component_useOnWindow_WXI9sB3m6wo", [
    hideTooltip
  ]));
  qwik.useVisibleTaskQrl(/* @__PURE__ */ qwik.inlinedQrl(({ track }) => {
    const [stateSignal2, update2] = qwik.useLexicalScope();
    const state = track(() => stateSignal2.value);
    if (state === "unpositioned")
      update2();
  }, "Tooltip_component_useVisibleTask_MxQUZYxM4x8", [
    stateSignal,
    update
  ]));
  return /* @__PURE__ */ qwik._jsxC(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ qwik._jsxC(Wrapper, {
        // need an id for the tooltip;
        "aria-describedby": `#${id}`,
        children: /* @__PURE__ */ qwik._jsxC(qwik.Slot, null, 3, "CG_0"),
        onBlur$: /* @__PURE__ */ qwik.inlinedQrl(() => {
          const [hideTooltip2] = qwik.useLexicalScope();
          return hideTooltip2();
        }, "Tooltip_component__Fragment_Wrapper_onBlur_UuCe1StpMkw", [
          hideTooltip
        ]),
        onFocus$: /* @__PURE__ */ qwik.inlinedQrl(() => {
          const [showTooltip2] = qwik.useLexicalScope();
          return showTooltip2();
        }, "Tooltip_component__Fragment_Wrapper_onFocus_Xt6DxIpnfrQ", [
          showTooltip
        ]),
        onMouseEnter$: /* @__PURE__ */ qwik.inlinedQrl(() => {
          const [showTooltip2] = qwik.useLexicalScope();
          return showTooltip2();
        }, "Tooltip_component__Fragment_Wrapper_onMouseEnter_Ct50X7n7U6E", [
          showTooltip
        ]),
        onMouseLeave$: /* @__PURE__ */ qwik.inlinedQrl(() => {
          const [hideTooltip2] = qwik.useLexicalScope();
          return hideTooltip2();
        }, "Tooltip_component__Fragment_Wrapper_onMouseLeave_oOqwq92vOlc", [
          hideTooltip
        ]),
        ref: triggerAnchor,
        style: "width: fit-content;",
        tabIndex: 0,
        [qwik._IMMUTABLE]: {
          "aria-describedby": qwik._IMMUTABLE,
          onBlur$: qwik._IMMUTABLE,
          onFocus$: qwik._IMMUTABLE,
          onMouseEnter$: qwik._IMMUTABLE,
          onMouseLeave$: qwik._IMMUTABLE,
          ref: qwik._IMMUTABLE,
          style: qwik._IMMUTABLE,
          tabIndex: qwik._IMMUTABLE
        }
      }, 1, "CG_1"),
      /* @__PURE__ */ qwik._jsxC(Wrapper, {
        class: `${stateSignal.value} ${props1.class || ""}`,
        id,
        onAnimationEnd$: /* @__PURE__ */ qwik.inlinedQrl(() => {
          const [positionSignal2, stateSignal2] = qwik.useLexicalScope();
          if (stateSignal2.value == "closing") {
            stateSignal2.value = "hidden";
            positionSignal2.value = {
              x: 0,
              y: 0
            };
          }
        }, "Tooltip_component__Fragment_Wrapper_onAnimationEnd_v1ENxh73TXA", [
          positionSignal,
          stateSignal
        ]),
        ref: tooltipAnchor,
        role: "tooltip",
        ...props1,
        get style() {
          return `--duration: ${props.durationMs ?? 100}ms;--x: ${positionSignal.value.x || 0}px;--y: ${positionSignal.value.y || 0}px;`;
        },
        get "data-state"() {
          return stateSignal.value;
        },
        children: qwik._fnSignal((p0) => p0.content, [
          props
        ], "p0.content"),
        [qwik._IMMUTABLE]: {
          "data-state": qwik._fnSignal((p0) => p0.value, [
            stateSignal
          ], "p0.value"),
          id: qwik._IMMUTABLE,
          onAnimationEnd$: qwik._IMMUTABLE,
          ref: qwik._IMMUTABLE,
          role: qwik._IMMUTABLE,
          style: qwik._fnSignal((p0, p1) => `--duration: ${p1.durationMs ?? 100}ms;--x: ${p0.value.x || 0}px;--y: ${p0.value.y || 0}px;`, [
            positionSignal,
            props
          ], "`--duration: ${p1.durationMs??100}ms;`+`--x: ${p0.value.x||0}px;`+`--y: ${p0.value.y||0}px;`")
        }
      }, 0, "CG_2")
    ]
  }, 1, "CG_3");
}, "Tooltip_component_EkDvuGlapJE"));
exports.Tooltip = Tooltip;
