// @qwik-ui/headless/components/tooltip/tooltip.qwik.mjs
import { componentQrl, inlinedQrl, _restProps, useStylesScopedQrl, useId, useSignal, useOnWindow, useLexicalScope, useVisibleTaskQrl, _jsxC, Slot, _IMMUTABLE, _fnSignal } from "@builder.io/qwik";
import { Fragment } from "@builder.io/qwik/jsx-runtime";
import { offset, computePosition } from "@floating-ui/dom";
import styles from "./tooltip.css.qwik.mjs";
const Tooltip = /* @__PURE__ */ componentQrl(/* @__PURE__ */ inlinedQrl((props) => {
  const props1 = _restProps(props, [
    "offset",
    "content",
    "position",
    "durationMs"
  ]);
  useStylesScopedQrl(/* @__PURE__ */ inlinedQrl(styles, "Tooltip_component_useStylesScoped_3jJQGcomGNw"));
  const id = useId();
  const triggerAnchor = useSignal();
  const tooltipAnchor = useSignal();
  const stateSignal = useSignal("closing");
  const positionSignal = useSignal({
    x: 0,
    y: 0
  });
  const Wrapper = props1.inline ? "span" : "div";
  const lastActivatedTimestamp = useSignal(Date.now());
  const update = /* @__PURE__ */ inlinedQrl(async () => {
    const [lastActivatedTimestamp2, positionSignal2, props2, stateSignal2, tooltipAnchor2, triggerAnchor2] = useLexicalScope();
    const now = Date.now();
    const middleware = [
      offset(props2.offset)
    ];
    const hasMouseEnterDebounced = now - lastActivatedTimestamp2.value >= 300;
    if (triggerAnchor2.value && tooltipAnchor2.value && hasMouseEnterDebounced) {
      const { x, y } = await computePosition(triggerAnchor2.value, tooltipAnchor2.value, {
        placement: props2.position ?? "top",
        middleware
      });
      lastActivatedTimestamp2.value = now;
      positionSignal2.value = {
        x,
        y
      };
      stateSignal2.value = "positioned";
    }
  }, "Tooltip_component_update_b7EpqFj07nQ", [
    lastActivatedTimestamp,
    positionSignal,
    props,
    stateSignal,
    tooltipAnchor,
    triggerAnchor
  ]);
  const showTooltip = /* @__PURE__ */ inlinedQrl(() => {
    const [props2, stateSignal2, update2] = useLexicalScope();
    update2();
    setTimeout(() => stateSignal2.value = "positioned", props2.durationMs ?? 100);
  }, "Tooltip_component_showTooltip_f0VZ50EMnCA", [
    props,
    stateSignal,
    update
  ]);
  const hideTooltip = /* @__PURE__ */ inlinedQrl(() => {
    const [props2, stateSignal2] = useLexicalScope();
    setTimeout(() => stateSignal2.value = "closing", props2.durationMs ?? 100);
  }, "Tooltip_component_hideTooltip_o30x7aVxT7s", [
    props,
    stateSignal
  ]);
  useOnWindow("keyup", /* @__PURE__ */ inlinedQrl((e) => {
    const [hideTooltip2] = useLexicalScope();
    const key = e.key;
    if (key === "Escape")
      hideTooltip2();
  }, "Tooltip_component_useOnWindow_WXI9sB3m6wo", [
    hideTooltip
  ]));
  useVisibleTaskQrl(/* @__PURE__ */ inlinedQrl(({ track }) => {
    const [stateSignal2, update2] = useLexicalScope();
    const state = track(() => stateSignal2.value);
    if (state === "unpositioned")
      update2();
  }, "Tooltip_component_useVisibleTask_MxQUZYxM4x8", [
    stateSignal,
    update
  ]));
  return /* @__PURE__ */ _jsxC(Fragment, {
    children: [
      /* @__PURE__ */ _jsxC(Wrapper, {
        // need an id for the tooltip;
        "aria-describedby": `#${id}`,
        children: /* @__PURE__ */ _jsxC(Slot, null, 3, "CG_0"),
        onBlur$: /* @__PURE__ */ inlinedQrl(() => {
          const [hideTooltip2] = useLexicalScope();
          return hideTooltip2();
        }, "Tooltip_component__Fragment_Wrapper_onBlur_UuCe1StpMkw", [
          hideTooltip
        ]),
        onFocus$: /* @__PURE__ */ inlinedQrl(() => {
          const [showTooltip2] = useLexicalScope();
          return showTooltip2();
        }, "Tooltip_component__Fragment_Wrapper_onFocus_Xt6DxIpnfrQ", [
          showTooltip
        ]),
        onMouseEnter$: /* @__PURE__ */ inlinedQrl(() => {
          const [showTooltip2] = useLexicalScope();
          return showTooltip2();
        }, "Tooltip_component__Fragment_Wrapper_onMouseEnter_Ct50X7n7U6E", [
          showTooltip
        ]),
        onMouseLeave$: /* @__PURE__ */ inlinedQrl(() => {
          const [hideTooltip2] = useLexicalScope();
          return hideTooltip2();
        }, "Tooltip_component__Fragment_Wrapper_onMouseLeave_oOqwq92vOlc", [
          hideTooltip
        ]),
        ref: triggerAnchor,
        style: "width: fit-content;",
        tabIndex: 0,
        [_IMMUTABLE]: {
          "aria-describedby": _IMMUTABLE,
          onBlur$: _IMMUTABLE,
          onFocus$: _IMMUTABLE,
          onMouseEnter$: _IMMUTABLE,
          onMouseLeave$: _IMMUTABLE,
          ref: _IMMUTABLE,
          style: _IMMUTABLE,
          tabIndex: _IMMUTABLE
        }
      }, 1, "CG_1"),
      /* @__PURE__ */ _jsxC(Wrapper, {
        class: `${stateSignal.value} ${props1.class || ""}`,
        id,
        onAnimationEnd$: /* @__PURE__ */ inlinedQrl(() => {
          const [positionSignal2, stateSignal2] = useLexicalScope();
          if (stateSignal2.value == "closing") {
            stateSignal2.value = "hidden";
            positionSignal2.value = {
              x: 0,
              y: 0
            };
          }
        }, "Tooltip_component__Fragment_Wrapper_onAnimationEnd_v1ENxh73TXA", [
          positionSignal,
          stateSignal
        ]),
        ref: tooltipAnchor,
        role: "tooltip",
        ...props1,
        get style() {
          return `--duration: ${props.durationMs ?? 100}ms;--x: ${positionSignal.value.x || 0}px;--y: ${positionSignal.value.y || 0}px;`;
        },
        get "data-state"() {
          return stateSignal.value;
        },
        children: _fnSignal((p0) => p0.content, [
          props
        ], "p0.content"),
        [_IMMUTABLE]: {
          "data-state": _fnSignal((p0) => p0.value, [
            stateSignal
          ], "p0.value"),
          id: _IMMUTABLE,
          onAnimationEnd$: _IMMUTABLE,
          ref: _IMMUTABLE,
          role: _IMMUTABLE,
          style: _fnSignal((p0, p1) => `--duration: ${p1.durationMs ?? 100}ms;--x: ${p0.value.x || 0}px;--y: ${p0.value.y || 0}px;`, [
            positionSignal,
            props
          ], "`--duration: ${p1.durationMs??100}ms;`+`--x: ${p0.value.x||0}px;`+`--y: ${p0.value.y||0}px;`")
        }
      }, 0, "CG_2")
    ]
  }, 1, "CG_3");
}, "Tooltip_component_EkDvuGlapJE"));
export {
  Tooltip
};
