// @qwik-ui/headless/components/checkbox/checkbox.qwik.mjs
import { componentQrl, inlinedQrl, _jsxS, _jsxC, Slot, _restProps, useLexicalScope, _fnSignal } from "@builder.io/qwik";
const Label = /* @__PURE__ */ componentQrl(/* @__PURE__ */ inlinedQrl(({ ...props }) => {
  return /* @__PURE__ */ _jsxS("label", {
    ...props,
    children: /* @__PURE__ */ _jsxC(Slot, null, 3, "Ig_0")
  }, null, 0, "Ig_1");
}, "Label_component_Pzm7AZJhfxk"));
const Root = /* @__PURE__ */ componentQrl(/* @__PURE__ */ inlinedQrl((props) => {
  const props1 = _restProps(props, [
    "checked",
    "disabled",
    "name",
    "ariaLabel",
    "id",
    "value",
    "tabIndex",
    "onChange"
  ]);
  return /* @__PURE__ */ _jsxS("input", {
    ...props1,
    onChange$: /* @__PURE__ */ inlinedQrl((event, element) => {
      const [props2] = useLexicalScope();
      if (props2.onChange)
        props2.onChange(event, element);
    }, "Root_component_input_onChange_1aPAS0tiahc", [
      props
    ])
  }, {
    "aria-checked": _fnSignal((p0) => p0.checked, [
      props
    ], "p0.checked"),
    "aria-label": _fnSignal((p0) => p0.ariaLabel, [
      props
    ], "p0.ariaLabel"),
    checked: _fnSignal((p0) => p0.checked, [
      props
    ], "p0.checked"),
    disabled: _fnSignal((p0) => p0.disabled, [
      props
    ], "p0.disabled"),
    id: _fnSignal((p0) => p0.id, [
      props
    ], "p0.id"),
    name: _fnSignal((p0) => p0.name, [
      props
    ], "p0.name"),
    role: "checkbox",
    tabIndex: _fnSignal((p0) => p0.tabIndex, [
      props
    ], "p0.tabIndex"),
    type: "checkbox",
    value: _fnSignal((p0) => p0.value, [
      props
    ], "p0.value")
  }, 0, "Ig_2");
}, "Root_component_kVvUFAgmRvA"));
export {
  Label,
  Root
};
