// @qwik-ui/headless/components/checkbox/checkbox.qwik.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const qwik = require("@builder.io/qwik");
const Label = /* @__PURE__ */ qwik.componentQrl(/* @__PURE__ */ qwik.inlinedQrl(({ ...props }) => {
  return /* @__PURE__ */ qwik._jsxS("label", {
    ...props,
    children: /* @__PURE__ */ qwik._jsxC(qwik.Slot, null, 3, "Ig_0")
  }, null, 0, "Ig_1");
}, "Label_component_Pzm7AZJhfxk"));
const Root = /* @__PURE__ */ qwik.componentQrl(/* @__PURE__ */ qwik.inlinedQrl((props) => {
  const props1 = qwik._restProps(props, [
    "checked",
    "disabled",
    "name",
    "ariaLabel",
    "id",
    "value",
    "tabIndex",
    "onChange"
  ]);
  return /* @__PURE__ */ qwik._jsxS("input", {
    ...props1,
    onChange$: /* @__PURE__ */ qwik.inlinedQrl((event, element) => {
      const [props2] = qwik.useLexicalScope();
      if (props2.onChange)
        props2.onChange(event, element);
    }, "Root_component_input_onChange_1aPAS0tiahc", [
      props
    ])
  }, {
    "aria-checked": qwik._fnSignal((p0) => p0.checked, [
      props
    ], "p0.checked"),
    "aria-label": qwik._fnSignal((p0) => p0.ariaLabel, [
      props
    ], "p0.ariaLabel"),
    checked: qwik._fnSignal((p0) => p0.checked, [
      props
    ], "p0.checked"),
    disabled: qwik._fnSignal((p0) => p0.disabled, [
      props
    ], "p0.disabled"),
    id: qwik._fnSignal((p0) => p0.id, [
      props
    ], "p0.id"),
    name: qwik._fnSignal((p0) => p0.name, [
      props
    ], "p0.name"),
    role: "checkbox",
    tabIndex: qwik._fnSignal((p0) => p0.tabIndex, [
      props
    ], "p0.tabIndex"),
    type: "checkbox",
    value: qwik._fnSignal((p0) => p0.value, [
      props
    ], "p0.value")
  }, 0, "Ig_2");
}, "Root_component_kVvUFAgmRvA"));
exports.Label = Label;
exports.Root = Root;
