// @qwik-ui/headless/components/menu/menu.qwik.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const qwik = require("@builder.io/qwik");
const MENU_CONTEXT_NAME = "qui-menu";
const quiMenuContext = qwik.createContextId(MENU_CONTEXT_NAME);
exports.KEYBOARD_KEY_NAME = void 0;
(function(KEYBOARD_KEY_NAME2) {
  KEYBOARD_KEY_NAME2["ARROW_UP"] = "ArrowUp";
  KEYBOARD_KEY_NAME2["ARROW_DOWN"] = "ArrowDown";
  KEYBOARD_KEY_NAME2["ESCAPE"] = "Escape";
})(exports.KEYBOARD_KEY_NAME || (exports.KEYBOARD_KEY_NAME = {}));
exports.CSS_CLASS_NAMES = void 0;
(function(CSS_CLASS_NAMES2) {
  CSS_CLASS_NAMES2["IS_EXPANDED"] = "quiIsExpanded";
  CSS_CLASS_NAMES2["IS_FOCUSED"] = "quiIsFocused";
  CSS_CLASS_NAMES2["IS_HOVERED"] = "quiIsHovered";
})(exports.CSS_CLASS_NAMES || (exports.CSS_CLASS_NAMES = {}));
const Menu = /* @__PURE__ */ qwik.componentQrl(/* @__PURE__ */ qwik.inlinedQrl((props) => {
  const parentId = qwik.useId();
  const childId = qwik.useId();
  const isExpanded = qwik.useSignal(props?.isExpanded || false);
  const isHovered = qwik.useSignal(false);
  const container = qwik.useSignal();
  const children = qwik.useStore([]);
  const currentButtonInFocusIndex = qwik.useSignal(-1);
  const triggerElementRef = qwik.useSignal();
  const currentId = qwik.useSignal("");
  const menuContextService = {
    currentId
  };
  qwik.useContextProvider(quiMenuContext, menuContextService);
  qwik.useVisibleTaskQrl(/* @__PURE__ */ qwik.inlinedQrl(({ track }) => {
    const [currentButtonInFocusIndex2, currentId2, isExpanded2] = qwik.useLexicalScope();
    track(() => isExpanded2.value);
    if (!isExpanded2.value) {
      currentButtonInFocusIndex2.value = -1;
      currentId2.value = "";
    }
  }, "Menu_component_useVisibleTask_z5TUX3lfqYo", [
    currentButtonInFocusIndex,
    currentId,
    isExpanded
  ]));
  qwik.useVisibleTaskQrl(/* @__PURE__ */ qwik.inlinedQrl(() => {
    const [children2, container2] = qwik.useLexicalScope();
    const options = container2.value?.querySelectorAll("button");
    if (options?.length)
      options.forEach((option) => children2.push(option));
  }, "Menu_component_useVisibleTask_1_nW9Eu00kceY", [
    children,
    container
  ]));
  qwik.useOnWindow("click", /* @__PURE__ */ qwik.inlinedQrl((event) => {
    const [isExpanded2, triggerElementRef2] = qwik.useLexicalScope();
    const target = event.target;
    if (isExpanded2.value && target !== triggerElementRef2.value)
      isExpanded2.value = false;
  }, "Menu_component_useOnWindow_ZOEh0cr01Zo", [
    isExpanded,
    triggerElementRef
  ]));
  return /* @__PURE__ */ qwik._jsxQ("div", {
    class: [
      props.class,
      {
        [exports.CSS_CLASS_NAMES.IS_EXPANDED]: isExpanded.value,
        [exports.CSS_CLASS_NAMES.IS_HOVERED]: isHovered.value
      }
    ]
  }, {
    id: parentId,
    onKeyDown$: /* @__PURE__ */ qwik.inlinedQrl((event) => {
      const [children2, currentButtonInFocusIndex2, currentId2, isExpanded2, triggerElementRef2] = qwik.useLexicalScope();
      if (event.key === exports.KEYBOARD_KEY_NAME.ESCAPE && isExpanded2.value) {
        isExpanded2.value = false;
        triggerElementRef2.value?.focus();
        return;
      }
      if (event.key === exports.KEYBOARD_KEY_NAME.ARROW_DOWN || event.key === exports.KEYBOARD_KEY_NAME.ARROW_UP) {
        let idx = currentButtonInFocusIndex2.value;
        if (event.key === exports.KEYBOARD_KEY_NAME.ARROW_DOWN) {
          idx = currentButtonInFocusIndex2.value + 1;
          if (idx >= children2.length)
            idx = 0;
        } else if (event.key === exports.KEYBOARD_KEY_NAME.ARROW_UP) {
          idx = currentButtonInFocusIndex2.value - 1;
          if (idx < 0)
            idx = children2.length - 1;
        }
        children2[idx].focus();
        currentId2.value = children2[idx].id;
        currentButtonInFocusIndex2.value = idx;
      }
    }, "Menu_component_div_onKeyDown_ImVFKNhBMvE", [
      children,
      currentButtonInFocusIndex,
      currentId,
      isExpanded,
      triggerElementRef
    ]),
    onMouseEnter$: /* @__PURE__ */ qwik.inlinedQrl(() => {
      const [isHovered2] = qwik.useLexicalScope();
      return isHovered2.value = true;
    }, "Menu_component_div_onMouseEnter_Un3HTfrqbP4", [
      isHovered
    ]),
    onMouseLeave$: /* @__PURE__ */ qwik.inlinedQrl(() => {
      const [isHovered2] = qwik.useLexicalScope();
      return isHovered2.value = false;
    }, "Menu_component_div_onMouseLeave_ehS07sC621w", [
      isHovered
    ])
  }, [
    /* @__PURE__ */ qwik._jsxQ("button", {
      ref: triggerElementRef
    }, {
      "aria-controls": qwik._fnSignal((p0, p1) => p1.value ? p0 : "", [
        childId,
        isExpanded
      ], 'p1.value?p0:""'),
      "aria-expanded": qwik._fnSignal((p0) => p0.value, [
        isExpanded
      ], "p0.value"),
      "aria-haspopup": true,
      onClick$: /* @__PURE__ */ qwik.inlinedQrl(() => {
        const [isExpanded2] = qwik.useLexicalScope();
        return isExpanded2.value = !isExpanded2.value;
      }, "Menu_component_div_button_onClick_s7AfHHK96OE", [
        isExpanded
      ])
    }, qwik._fnSignal((p0) => p0.triggerElement || "Menu", [
      props
    ], 'p0.triggerElement||"Menu"'), 3, null),
    /* @__PURE__ */ qwik._jsxQ("nav", {
      ref: container
    }, {
      "aria-labelledby": parentId,
      id: childId,
      role: "menu",
      style: qwik._fnSignal((p0) => ({
        visibility: p0.value ? "visible" : "hidden"
      }), [
        isExpanded
      ], '{visibility:p0.value?"visible":"hidden"}')
    }, /* @__PURE__ */ qwik._jsxC(qwik.Slot, null, 3, "nG_0"), 1, null)
  ], 1, "nG_1");
}, "Menu_component_GMwdTIoT1z4"));
const MenuItem = /* @__PURE__ */ qwik.componentQrl(/* @__PURE__ */ qwik.inlinedQrl((props) => {
  qwik._jsxBranch();
  const myId = qwik.useId();
  const contextService = qwik.useContext(quiMenuContext);
  const isFocused = contextService.currentId.value === myId;
  return /* @__PURE__ */ qwik._jsxQ("div", null, {
    class: qwik._fnSignal((p0) => p0.class, [
      props
    ], "p0.class")
  }, props.onClick$ ? /* @__PURE__ */ qwik._jsxQ("button", {
    class: {
      [exports.CSS_CLASS_NAMES.IS_FOCUSED]: isFocused
    },
    onClick$: props.onClick$
  }, {
    disabled: qwik._fnSignal((p0) => p0.disabled || false, [
      props
    ], "p0.disabled||false"),
    id: myId,
    tabIndex: -1
  }, /* @__PURE__ */ qwik._jsxC(qwik.Slot, null, 3, "nG_2"), 0, "nG_3") : /* @__PURE__ */ qwik._jsxC(qwik.Slot, null, 3, "nG_4"), 1, "nG_5");
}, "MenuItem_component_Wd5FkIK10wk"));
exports.Menu = Menu;
exports.MenuItem = MenuItem;
exports.quiMenuContext = quiMenuContext;
