// @qwik-ui/headless/components/accordion/accordion-root.qwik.mjs
import { componentQrl, inlinedQrl, _restProps, useSignal, useTaskQrl, useLexicalScope, useVisibleTaskQrl, useContextProvider, _jsxS, _jsxC, Slot } from "@builder.io/qwik";
import { accordionRootContextId } from "./accordion-context-id.qwik.mjs";
const AccordionRoot = /* @__PURE__ */ componentQrl(/* @__PURE__ */ inlinedQrl((props) => {
  const props1 = _restProps(props, [
    "collapsible",
    "behavior",
    "animated",
    "onSelectedIndexChange$",
    "onFocusIndexChange$"
  ]);
  const rootRef = useSignal();
  const currFocusedTriggerIndexSig = useSignal(-1);
  const currSelectedTriggerIndexSig = useSignal(-1);
  const selectedTriggerIdSig = useSignal("");
  const triggerElementsSig = useSignal([]);
  useTaskQrl(/* @__PURE__ */ inlinedQrl(({ track }) => {
    const [currSelectedTriggerIndexSig2, props2] = useLexicalScope();
    track(() => currSelectedTriggerIndexSig2.value);
    if (props2.onSelectedIndexChange$)
      props2.onSelectedIndexChange$(currSelectedTriggerIndexSig2.value);
  }, "AccordionRoot_component_useTask_Qwqwbzx0Lr8", [
    currSelectedTriggerIndexSig,
    props
  ]));
  useTaskQrl(/* @__PURE__ */ inlinedQrl(({ track }) => {
    const [currFocusedTriggerIndexSig2, props2] = useLexicalScope();
    track(() => currFocusedTriggerIndexSig2.value);
    if (props2.onFocusIndexChange$)
      props2.onFocusIndexChange$(currFocusedTriggerIndexSig2.value);
  }, "AccordionRoot_component_useTask_1_eCYTRyvWCIU", [
    currFocusedTriggerIndexSig,
    props
  ]));
  const updateTriggers$ = /* @__PURE__ */ inlinedQrl(() => {
    const [rootRef2, triggerElementsSig2] = useLexicalScope();
    if (!rootRef2.value)
      return;
    const getLatestTriggers = Array.from(rootRef2.value.querySelectorAll("[data-trigger-id]"));
    triggerElementsSig2.value = getLatestTriggers.filter((element) => {
      if (element.getAttribute("aria-disabled") === "true")
        return false;
      return true;
    });
  }, "AccordionRoot_component_updateTriggers_IvCdCytWFQ0", [
    rootRef,
    triggerElementsSig
  ]);
  const focusPreviousTrigger$ = /* @__PURE__ */ inlinedQrl(() => {
    const [currFocusedTriggerIndexSig2, triggerElementsSig2] = useLexicalScope();
    if (currFocusedTriggerIndexSig2.value === 0) {
      currFocusedTriggerIndexSig2.value = triggerElementsSig2.value.length - 1;
      return triggerElementsSig2.value[triggerElementsSig2.value.length - 1].focus();
    }
    currFocusedTriggerIndexSig2.value--;
    return triggerElementsSig2.value[currFocusedTriggerIndexSig2.value].focus();
  }, "AccordionRoot_component_focusPreviousTrigger_ImTP7x2Qm0M", [
    currFocusedTriggerIndexSig,
    triggerElementsSig
  ]);
  const focusNextTrigger$ = /* @__PURE__ */ inlinedQrl(() => {
    const [currFocusedTriggerIndexSig2, triggerElementsSig2] = useLexicalScope();
    if (currFocusedTriggerIndexSig2.value === triggerElementsSig2.value.length - 1) {
      currFocusedTriggerIndexSig2.value = 0;
      return triggerElementsSig2.value[0].focus();
    }
    currFocusedTriggerIndexSig2.value++;
    return triggerElementsSig2.value[currFocusedTriggerIndexSig2.value].focus();
  }, "AccordionRoot_component_focusNextTrigger_9te8634Gk34", [
    currFocusedTriggerIndexSig,
    triggerElementsSig
  ]);
  const focusFirstTrigger$ = /* @__PURE__ */ inlinedQrl(() => {
    const [triggerElementsSig2] = useLexicalScope();
    return triggerElementsSig2.value[0].focus();
  }, "AccordionRoot_component_focusFirstTrigger_68n0b3BkBYI", [
    triggerElementsSig
  ]);
  const focusLastTrigger$ = /* @__PURE__ */ inlinedQrl(() => {
    const [triggerElementsSig2] = useLexicalScope();
    return triggerElementsSig2.value[triggerElementsSig2.value.length - 1].focus();
  }, "AccordionRoot_component_focusLastTrigger_i9ed6fSgwd0", [
    triggerElementsSig
  ]);
  useVisibleTaskQrl(/* @__PURE__ */ inlinedQrl(function reIndexTriggers() {
    const [updateTriggers$2] = useLexicalScope();
    updateTriggers$2();
  }, "AccordionRoot_component_useVisibleTask_c7PiYJ1s0GU", [
    updateTriggers$
  ]));
  const contextService = {
    updateTriggers$,
    focusFirstTrigger$,
    focusPreviousTrigger$,
    focusNextTrigger$,
    focusLastTrigger$,
    currFocusedTriggerIndexSig,
    currSelectedTriggerIndexSig,
    selectedTriggerIdSig,
    triggerElementsSig,
    collapsible: props.collapsible ?? true,
    behavior: props.behavior ?? "single",
    animated: props.animated ?? false
  };
  useContextProvider(accordionRootContextId, contextService);
  return /* @__PURE__ */ _jsxS("div", {
    ...props1,
    children: /* @__PURE__ */ _jsxC(Slot, null, 3, "jW_0"),
    ref: rootRef
  }, null, 0, "jW_1");
}, "AccordionRoot_component_dyqeXuV0Jjc"));
export {
  AccordionRoot
};
