// @qwik-ui/headless/components/tabs/tab-panel.qwik.mjs
import { componentQrl, inlinedQrl, _restProps, useContext, useComputedQrl, useLexicalScope, _jsxS, _jsxC, Slot, _fnSignal } from "@builder.io/qwik";
import { TAB_ID_PREFIX } from "./tab.qwik.mjs";
import { tabsContextId } from "./tabs-context-id.qwik.mjs";
const TAB_PANEL_ID_PREFIX = "_tabpanel_";
const TabPanel = /* @__PURE__ */ componentQrl(/* @__PURE__ */ inlinedQrl(
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  (props) => {
    const props1 = _restProps(props, [
      "label",
      "_tabId",
      "_extraClass"
    ]);
    const contextService = useContext(tabsContextId);
    const fullPanelElementId = contextService.tabsPrefix + TAB_PANEL_ID_PREFIX + props._tabId;
    const fullTabElementId = contextService.tabsPrefix + TAB_ID_PREFIX + props._tabId;
    const isSelectedSig = useComputedQrl(/* @__PURE__ */ inlinedQrl(() => {
      const [contextService2, props2] = useLexicalScope();
      return contextService2.selectedTabIdSig.value === props2._tabId;
    }, "TabPanel_component_isSelectedSig_useComputed_i0H9g0nHXC0", [
      contextService,
      props
    ]));
    return /* @__PURE__ */ _jsxS("div", {
      ...props1,
      "aria-labelledby": fullTabElementId,
      children: /* @__PURE__ */ _jsxC(Slot, null, 3, "EC_0"),
      class: [
        props1.class?.value ?? props1.class,
        props._extraClass?.value ?? props._extraClass
      ],
      id: fullPanelElementId
    }, {
      hidden: _fnSignal((p0) => !p0.value, [
        isSelectedSig
      ], "!p0.value"),
      role: "tabpanel",
      tabIndex: 0
    }, 0, "EC_1");
  },
  "TabPanel_component_3rcWDF00mPE"
));
export {
  TAB_PANEL_ID_PREFIX,
  TabPanel
};
