// @qwik-ui/headless/components/tabs/tab-panel.qwik.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const qwik = require("@builder.io/qwik");
const tab = require("./tab.qwik.cjs");
const tabsContextId = require("./tabs-context-id.qwik.cjs");
const TAB_PANEL_ID_PREFIX = "_tabpanel_";
const TabPanel = /* @__PURE__ */ qwik.componentQrl(/* @__PURE__ */ qwik.inlinedQrl(
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  (props) => {
    const props1 = qwik._restProps(props, [
      "label",
      "_tabId",
      "_extraClass"
    ]);
    const contextService = qwik.useContext(tabsContextId.tabsContextId);
    const fullPanelElementId = contextService.tabsPrefix + TAB_PANEL_ID_PREFIX + props._tabId;
    const fullTabElementId = contextService.tabsPrefix + tab.TAB_ID_PREFIX + props._tabId;
    const isSelectedSig = qwik.useComputedQrl(/* @__PURE__ */ qwik.inlinedQrl(() => {
      const [contextService2, props2] = qwik.useLexicalScope();
      return contextService2.selectedTabIdSig.value === props2._tabId;
    }, "TabPanel_component_isSelectedSig_useComputed_i0H9g0nHXC0", [
      contextService,
      props
    ]));
    return /* @__PURE__ */ qwik._jsxS("div", {
      ...props1,
      "aria-labelledby": fullTabElementId,
      children: /* @__PURE__ */ qwik._jsxC(qwik.Slot, null, 3, "EC_0"),
      class: [
        props1.class?.value ?? props1.class,
        props._extraClass?.value ?? props._extraClass
      ],
      id: fullPanelElementId
    }, {
      hidden: qwik._fnSignal((p0) => !p0.value, [
        isSelectedSig
      ], "!p0.value"),
      role: "tabpanel",
      tabIndex: 0
    }, 0, "EC_1");
  },
  "TabPanel_component_3rcWDF00mPE"
));
exports.TAB_PANEL_ID_PREFIX = TAB_PANEL_ID_PREFIX;
exports.TabPanel = TabPanel;
