// @qwik-ui/headless/components/modal/modal.qwik.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const qwik = require("@builder.io/qwik");
const modalBehavior = require("./modal-behavior.qwik.cjs");
const bodyScrollLock = require("body-scroll-lock");
const modal = require("./modal.css.qwik.cjs");
const Modal = /* @__PURE__ */ qwik.componentQrl(/* @__PURE__ */ qwik.inlinedQrl((props) => {
  qwik.useStylesQrl(/* @__PURE__ */ qwik.inlinedQrl(modal, "Modal_component_useStyles_6kc6yEJKobw"));
  const modalRefSig = qwik.useSignal();
  const { "bind:show": showSig } = props;
  qwik.useTaskQrl(/* @__PURE__ */ qwik.inlinedQrl(async function toggleModal({ track, cleanup }) {
    const [modalRefSig2, props2, showSig2] = qwik.useLexicalScope();
    const isOpen = track(() => showSig2.value);
    const modal2 = modalRefSig2.value;
    if (!modal2)
      return;
    const focusTrap = modalBehavior.trapFocus(modal2);
    window.addEventListener("keydown", modalBehavior.overrideNativeDialogEscapeBehaviorWith(() => showSig2.value = false), {
      once: true
    });
    if (isOpen) {
      modalBehavior.showModal(modal2);
      bodyScrollLock.disableBodyScroll(modal2, {
        reserveScrollBarGap: true
      });
      props2.onShow$?.();
      modalBehavior.activateFocusTrap(focusTrap);
    } else {
      modalBehavior.closeModal(modal2);
      props2.onClose$?.();
    }
    cleanup(() => {
      modalBehavior.deactivateFocusTrap(focusTrap);
    });
  }, "Modal_component_useTask_boBpVEMpnGQ", [
    modalRefSig,
    props,
    showSig
  ]));
  const closeOnBackdropClick$ = /* @__PURE__ */ qwik.inlinedQrl((event) => {
    const [modalRefSig2, props2, showSig2] = qwik.useLexicalScope();
    if (props2.alert === true || props2.closeOnBackdropClick === false)
      return;
    if (modalBehavior.wasModalBackdropClicked(modalRefSig2.value, event))
      showSig2.value = false;
  }, "Modal_component_closeOnBackdropClick_qbpjydAbzcE", [
    modalRefSig,
    props,
    showSig
  ]);
  return /* @__PURE__ */ qwik._jsxS("dialog", {
    ...props,
    children: /* @__PURE__ */ qwik._jsxC(qwik.Slot, null, 3, "9p_0"),
    onClick$: /* @__PURE__ */ qwik.inlinedQrl((event) => {
      const [closeOnBackdropClick$2] = qwik.useLexicalScope();
      return closeOnBackdropClick$2(event);
    }, "Modal_component_dialog_onClick_rnAxN4ftyEs", [
      closeOnBackdropClick$
    ]),
    ref: modalRefSig
  }, {
    role: qwik._fnSignal((p0) => p0.alert === true ? "alertdialog" : "dialog", [
      props
    ], 'p0.alert===true?"alertdialog":"dialog"')
  }, 0, "9p_1");
}, "Modal_component_VXOY7yQNLIw"));
exports.Modal = Modal;
