// @qwik-ui/headless/components/popover/popover.qwik.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const qwik = require("@builder.io/qwik");
const build = require("@builder.io/qwik/build");
const dom = require("@floating-ui/dom");
const popoverContext = require("./popover-context.qwik.cjs");
const Popover = /* @__PURE__ */ qwik.componentQrl(/* @__PURE__ */ qwik.inlinedQrl((props) => {
  const restOfProps = qwik._restProps(props, [
    "triggerEvent",
    "onUpdate$",
    "disableClickOutSide",
    "offset",
    "placement",
    "isOpen"
  ]);
  const wrapperRef = qwik.useSignal();
  const triggerRef = qwik.useSignal();
  const contentRef = qwik.useSignal();
  const setOverlayRef$ = /* @__PURE__ */ qwik.inlinedQrl((ref) => {
    const [contentRef2] = qwik.useLexicalScope();
    if (ref)
      contentRef2.value = ref.value;
  }, "Popover_component_setOverlayRef_0udWfF7cTAI", [
    contentRef
  ]);
  const setTriggerRef$ = /* @__PURE__ */ qwik.inlinedQrl((ref) => {
    const [triggerRef2] = qwik.useLexicalScope();
    if (ref)
      triggerRef2.value = ref.value;
  }, "Popover_component_setTriggerRef_GUXkt8iP7ik", [
    triggerRef
  ]);
  qwik.useStylesScopedQrl(/* @__PURE__ */ qwik.inlinedQrl(`
     [data-type="popover-root"] {
      position: relative;
     }
  `, "Popover_component_useStylesScoped_b2mXQfsnmBE"));
  const contextService = qwik.useStore({
    isOpen: false,
    triggerEvent: props.triggerEvent ?? "click",
    setTriggerRef$,
    setOverlayRef$
  });
  qwik.useContextProvider(popoverContext.PopoverContext, contextService);
  const closePopover = /* @__PURE__ */ qwik.inlinedQrl(async () => {
    const [contentRef2, contextService2, props2] = qwik.useLexicalScope();
    contextService2.isOpen = false;
    if (contentRef2) {
      contentRef2.value?.classList.add("close");
      contentRef2.value?.classList.remove("open");
    }
    if (props2.onUpdate$)
      await props2.onUpdate$(contextService2.isOpen);
  }, "Popover_component_closePopover_k1Gil9i90JI", [
    contentRef,
    contextService,
    props
  ]);
  const openPopover = /* @__PURE__ */ qwik.inlinedQrl(async () => {
    const [contentRef2, contextService2, props2] = qwik.useLexicalScope();
    contextService2.isOpen = true;
    if (contentRef2) {
      contentRef2.value?.classList.add("open");
      contentRef2.value?.classList.remove("close");
    }
    if (props2.onUpdate$)
      await props2.onUpdate$(contextService2.isOpen);
  }, "Popover_component_openPopover_gL5gEsnFqdE", [
    contentRef,
    contextService,
    props
  ]);
  const togglePopover = /* @__PURE__ */ qwik.inlinedQrl(async () => {
    const [closePopover2, contextService2, openPopover2, props2] = qwik.useLexicalScope();
    if (contextService2.isOpen)
      closePopover2();
    else
      openPopover2();
    if (props2.onUpdate$)
      await props2.onUpdate$(contextService2.isOpen);
  }, "Popover_component_togglePopover_wUcsAC2CDPs", [
    closePopover,
    contextService,
    openPopover,
    props
  ]);
  qwik.useTaskQrl(/* @__PURE__ */ qwik.inlinedQrl(({ track }) => {
    const [contentRef2, openPopover2, props2, triggerRef2] = qwik.useLexicalScope();
    const trigger = track(() => triggerRef2.value);
    const content = track(() => contentRef2.value);
    if (build.isBrowser && trigger && content) {
      dom.autoUpdate(trigger, content, () => {
        dom.computePosition(trigger, content, {
          middleware: [
            dom.flip(),
            dom.shift(),
            dom.offset(props2.offset || 0)
          ],
          placement: props2.placement
        }).then(({ x, y }) => {
          Object.assign(content.style, {
            left: `${x}px`,
            top: `${y}px`
          });
        });
      });
      if (props2.isOpen)
        openPopover2();
    }
  }, "Popover_component_useTask_kpkTN6NP76Y", [
    contentRef,
    openPopover,
    props,
    triggerRef
  ]));
  qwik.useVisibleTaskQrl(/* @__PURE__ */ qwik.inlinedQrl(({ track }) => {
    const [contextService2, props2] = qwik.useLexicalScope();
    track(() => props2.isOpen);
    contextService2.isOpen = !!props2.isOpen;
  }, "Popover_component_useVisibleTask_hP4F7a954yg", [
    contextService,
    props
  ]));
  qwik.useVisibleTaskQrl(/* @__PURE__ */ qwik.inlinedQrl(({ track }) => {
    const [closePopover2, contentRef2, contextService2, openPopover2, triggerRef2] = qwik.useLexicalScope();
    track(() => contextService2.isOpen);
    if (!triggerRef2.value || !contentRef2.value)
      return;
    if (contextService2.isOpen)
      openPopover2();
    else
      closePopover2();
  }, "Popover_component_useVisibleTask_1_iWRr6q6c5EQ", [
    closePopover,
    contentRef,
    contextService,
    openPopover,
    triggerRef
  ]));
  const clickHandler = /* @__PURE__ */ qwik.inlinedQrl((e) => {
    const [closePopover2, contentRef2, props2, togglePopover2, triggerRef2] = qwik.useLexicalScope();
    const isContentClicked = contentRef2.value?.contains(e.target);
    if (isContentClicked)
      return;
    const isTriggerClicked = triggerRef2.value?.contains(e.target);
    if (isTriggerClicked && (props2.triggerEvent ?? "click") === "click")
      togglePopover2();
    else {
      if (props2.disableClickOutSide)
        return;
      closePopover2();
    }
  }, "Popover_component_clickHandler_sDATyKm0S3o", [
    closePopover,
    contentRef,
    props,
    togglePopover,
    triggerRef
  ]);
  return /* @__PURE__ */ qwik._jsxS("span", {
    ...restOfProps,
    children: /* @__PURE__ */ qwik._jsxC(qwik.Slot, null, 3, "ca_0"),
    "document:onClick$": clickHandler,
    ref: wrapperRef
  }, {
    "data-type": "popover-root"
  }, 0, "ca_1");
}, "Popover_component_GIDbxXNx2UI"));
exports.Popover = Popover;
