// @qwik-ui/headless/components/select/select-root.qwik.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const qwik = require("@builder.io/qwik");
const dom = require("@floating-ui/dom");
const visuallyHidden = require("../../utils/visually-hidden.qwik.cjs");
const selectContextId = require("./select-context-id.qwik.cjs");
const selectNativeSelect = require("./select-native-select.qwik.cjs");
const SelectRoot = /* @__PURE__ */ qwik.componentQrl(/* @__PURE__ */ qwik.inlinedQrl((props) => {
  qwik._jsxBranch();
  const rootRefSig = qwik.useSignal();
  const optionsStore = qwik.useStore([]);
  const selectedOptionSig = qwik.useSignal("");
  const isOpenSig = qwik.useSignal(false);
  const triggerRefSig = qwik.useSignal();
  const listBoxRefSig = qwik.useSignal();
  const isListboxHiddenSig = qwik.useSignal(true);
  const selectContext = {
    optionsStore,
    selectedOptionSig,
    isOpenSig,
    triggerRefSig,
    listBoxRefSig,
    isListboxHiddenSig
  };
  qwik.useContextProvider(selectContextId, selectContext);
  qwik.useOnDocument("click", /* @__PURE__ */ qwik.inlinedQrl((e) => {
    const [rootRefSig2, selectContext2] = qwik.useLexicalScope();
    const target = e.target;
    if (selectContext2.isOpenSig.value === true && !rootRefSig2.value?.contains(target))
      selectContext2.isOpenSig.value = false;
  }, "SelectRoot_component_useOnDocument_COnj037onYs", [
    rootRefSig,
    selectContext
  ]));
  qwik.useVisibleTaskQrl(/* @__PURE__ */ qwik.inlinedQrl(function setKeyHandler({ cleanup }) {
    const [rootRefSig2, selectContext2] = qwik.useLexicalScope();
    function keyHandler(e) {
      e.preventDefault();
      if (e.key === "Escape")
        selectContext2.isOpenSig.value = false;
    }
    rootRefSig2.value?.addEventListener("keydown", keyHandler);
    cleanup(() => {
      rootRefSig2.value?.removeEventListener("keydown", keyHandler);
    });
  }, "SelectRoot_component_useVisibleTask_pNx9pkBC1tI", [
    rootRefSig,
    selectContext
  ]));
  const updatePosition$ = /* @__PURE__ */ qwik.inlinedQrl((referenceEl, floatingEl) => {
    dom.computePosition(referenceEl, floatingEl, {
      placement: "bottom",
      middleware: [
        dom.flip()
      ]
    }).then(({ x, y }) => {
      Object.assign(floatingEl.style, {
        left: `${x}px`,
        top: `${y}px`
      });
    });
  }, "SelectRoot_component_updatePosition_Bp5svGmsvFc");
  qwik.useVisibleTaskQrl(/* @__PURE__ */ qwik.inlinedQrl(async function toggleSelectListBox({ track }) {
    const [isListboxHiddenSig2, selectContext2, updatePosition$2] = qwik.useLexicalScope();
    const trigger = track(() => selectContext2.triggerRefSig.value);
    const listBox = track(() => selectContext2.listBoxRefSig.value);
    const expanded = track(() => selectContext2.isOpenSig.value);
    if (!trigger || !listBox)
      return;
    if (expanded === true) {
      listBox.style.visibility = "hidden";
      await updatePosition$2(trigger, listBox);
      listBox.style.visibility = "visible";
      isListboxHiddenSig2.value = false;
      listBox?.focus();
    }
    if (expanded === false)
      trigger?.focus();
  }, "SelectRoot_component_useVisibleTask_1_N7H39giT6pA", [
    isListboxHiddenSig,
    selectContext,
    updatePosition$
  ]));
  qwik.useVisibleTaskQrl(/* @__PURE__ */ qwik.inlinedQrl(function collectOptions({ track }) {
    const [selectContext2] = qwik.useLexicalScope();
    const listBox = track(() => selectContext2.listBoxRefSig.value);
    if (listBox) {
      const collectedOptions = Array.from(listBox.querySelectorAll('[role="option"]'));
      selectContext2.optionsStore.push(...collectedOptions);
    }
  }, "SelectRoot_component_useVisibleTask_2_tytKysVfH9M", [
    selectContext
  ]));
  return /* @__PURE__ */ qwik._jsxS("div", {
    ref: rootRefSig,
    ...props,
    children: [
      /* @__PURE__ */ qwik._jsxC(qwik.Slot, null, 3, "Bs_0"),
      props.required ? /* @__PURE__ */ qwik._jsxC(visuallyHidden.VisuallyHidden, {
        children: /* @__PURE__ */ qwik._jsxC(selectNativeSelect.NativeSelect, null, 3, "Bs_1")
      }, 1, "Bs_2") : null
    ]
  }, null, 0, "Bs_3");
}, "SelectRoot_component_1P8uqC3afXw"));
exports.SelectRoot = SelectRoot;
