// @qwik-ui/headless/components/select/select-listbox.qwik.mjs
import { componentQrl, inlinedQrl, useSignal, useContext, useVisibleTaskQrl, useLexicalScope, _jsxQ, _fnSignal, _jsxC, Slot } from "@builder.io/qwik";
import SelectContextId from "./select-context-id.qwik.mjs";
const SelectListBox = /* @__PURE__ */ componentQrl(/* @__PURE__ */ inlinedQrl((props) => {
  const listBoxRef = useSignal();
  const selectContext = useContext(SelectContextId);
  selectContext.listBoxRefSig = listBoxRef;
  useVisibleTaskQrl(/* @__PURE__ */ inlinedQrl(function setKeyHandler({ cleanup }) {
    const [listBoxRef2, selectContext2] = useLexicalScope();
    function keyHandler(e) {
      const availableOptions = selectContext2.optionsStore.filter((option) => !(option?.getAttribute("aria-disabled") === "true"));
      const target = e.target;
      const currentIndex = availableOptions.indexOf(target);
      if (e.key === "ArrowDown" || e.key === "ArrowUp" || e.key === "Home" || e.key === "End" || e.key === " ")
        e.preventDefault();
      if (e.key === "ArrowDown") {
        if (currentIndex === availableOptions.length - 1)
          availableOptions[0]?.focus();
        else
          availableOptions[currentIndex + 1]?.focus();
      }
      if (e.key === "ArrowUp") {
        if (currentIndex <= 0)
          availableOptions[availableOptions.length - 1]?.focus();
        else
          availableOptions[currentIndex - 1]?.focus();
      }
    }
    listBoxRef2.value?.addEventListener("keydown", keyHandler);
    cleanup(() => {
      listBoxRef2.value?.removeEventListener("keydown", keyHandler);
    });
  }, "SelectListBox_component_useVisibleTask_NZcLPc4zmDI", [
    listBoxRef,
    selectContext
  ]));
  return /* @__PURE__ */ _jsxQ("ul", {
    ref: listBoxRef
  }, {
    class: _fnSignal((p0) => p0.class, [
      props
    ], "p0.class"),
    hidden: _fnSignal((p0) => !p0.isListboxHiddenSig.value, [
      selectContext
    ], "!p0.isListboxHiddenSig.value"),
    role: "listbox",
    style: _fnSignal((p0, p1) => `
        display: ${p1.isOpenSig.value ? "block" : "none"};
        position: absolute;
        z-index: 1;
        ${p0.style}
      `, [
      props,
      selectContext
    ], '`\n        display: ${p1.isOpenSig.value?"block":"none"};\n        position: absolute;\n        z-index: 1;\n        ${p0.style}\n      `'),
    tabIndex: 0
  }, /* @__PURE__ */ _jsxC(Slot, null, 3, "bD_0"), 1, "bD_1");
}, "SelectListBox_component_DIywLbk0fWc"));
export {
  SelectListBox
};
