// @qwik-ui/headless/components/select/select-listbox.qwik.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const qwik = require("@builder.io/qwik");
const selectContextId = require("./select-context-id.qwik.cjs");
const SelectListBox = /* @__PURE__ */ qwik.componentQrl(/* @__PURE__ */ qwik.inlinedQrl((props) => {
  const listBoxRef = qwik.useSignal();
  const selectContext = qwik.useContext(selectContextId);
  selectContext.listBoxRefSig = listBoxRef;
  qwik.useVisibleTaskQrl(/* @__PURE__ */ qwik.inlinedQrl(function setKeyHandler({ cleanup }) {
    const [listBoxRef2, selectContext2] = qwik.useLexicalScope();
    function keyHandler(e) {
      const availableOptions = selectContext2.optionsStore.filter((option) => !(option?.getAttribute("aria-disabled") === "true"));
      const target = e.target;
      const currentIndex = availableOptions.indexOf(target);
      if (e.key === "ArrowDown" || e.key === "ArrowUp" || e.key === "Home" || e.key === "End" || e.key === " ")
        e.preventDefault();
      if (e.key === "ArrowDown") {
        if (currentIndex === availableOptions.length - 1)
          availableOptions[0]?.focus();
        else
          availableOptions[currentIndex + 1]?.focus();
      }
      if (e.key === "ArrowUp") {
        if (currentIndex <= 0)
          availableOptions[availableOptions.length - 1]?.focus();
        else
          availableOptions[currentIndex - 1]?.focus();
      }
    }
    listBoxRef2.value?.addEventListener("keydown", keyHandler);
    cleanup(() => {
      listBoxRef2.value?.removeEventListener("keydown", keyHandler);
    });
  }, "SelectListBox_component_useVisibleTask_NZcLPc4zmDI", [
    listBoxRef,
    selectContext
  ]));
  return /* @__PURE__ */ qwik._jsxQ("ul", {
    ref: listBoxRef
  }, {
    class: qwik._fnSignal((p0) => p0.class, [
      props
    ], "p0.class"),
    hidden: qwik._fnSignal((p0) => !p0.isListboxHiddenSig.value, [
      selectContext
    ], "!p0.isListboxHiddenSig.value"),
    role: "listbox",
    style: qwik._fnSignal((p0, p1) => `
        display: ${p1.isOpenSig.value ? "block" : "none"};
        position: absolute;
        z-index: 1;
        ${p0.style}
      `, [
      props,
      selectContext
    ], '`\n        display: ${p1.isOpenSig.value?"block":"none"};\n        position: absolute;\n        z-index: 1;\n        ${p0.style}\n      `'),
    tabIndex: 0
  }, /* @__PURE__ */ qwik._jsxC(qwik.Slot, null, 3, "bD_0"), 1, "bD_1");
}, "SelectListBox_component_DIywLbk0fWc"));
exports.SelectListBox = SelectListBox;
