// @qwik-ui/headless/components/select/select-trigger.qwik.mjs
import { componentQrl, inlinedQrl, useContext, useSignal, useVisibleTaskQrl, useLexicalScope, _jsxS, _jsxC, Slot, _fnSignal } from "@builder.io/qwik";
import SelectContextId from "./select-context-id.qwik.mjs";
const SelectTrigger = /* @__PURE__ */ componentQrl(/* @__PURE__ */ inlinedQrl((props) => {
  const selectContext = useContext(SelectContextId);
  const triggerRef = useSignal();
  selectContext.triggerRefSig = triggerRef;
  useVisibleTaskQrl(/* @__PURE__ */ inlinedQrl(function setClickHandler({ cleanup }) {
    const [selectContext2, triggerRef2] = useLexicalScope();
    function clickHandler(e) {
      e.preventDefault();
      selectContext2.isOpenSig.value = !selectContext2.isOpenSig.value;
    }
    triggerRef2.value?.addEventListener("click", clickHandler);
    cleanup(() => {
      triggerRef2.value?.removeEventListener("click", clickHandler);
    });
  }, "SelectTrigger_component_useVisibleTask_80qoWIXMIfc", [
    selectContext,
    triggerRef
  ]));
  useVisibleTaskQrl(/* @__PURE__ */ inlinedQrl(function setKeyHandler({ cleanup }) {
    const [selectContext2, triggerRef2] = useLexicalScope();
    function keyHandler(e) {
      if (e.key === "Home" || e.key === "End")
        e.preventDefault();
      if (e.key === "ArrowDown" || e.key === "ArrowUp" || e.key === "Enter" || e.key === " ")
        selectContext2.isOpenSig.value = true;
    }
    triggerRef2.value?.addEventListener("keydown", keyHandler);
    cleanup(() => {
      triggerRef2.value?.removeEventListener("keydown", keyHandler);
    });
  }, "SelectTrigger_component_useVisibleTask_1_wQEhBlyku50", [
    selectContext,
    triggerRef
  ]));
  return /* @__PURE__ */ _jsxS("button", {
    ref: triggerRef,
    ...props,
    children: /* @__PURE__ */ _jsxC(Slot, null, 3, "VR_0")
  }, {
    "aria-expanded": _fnSignal((p0) => p0.isOpenSig.value, [
      selectContext
    ], "p0.isOpenSig.value")
  }, 0, "VR_1");
}, "SelectTrigger_component_QNah0rGW5hg"));
export {
  SelectTrigger
};
