// @qwik-ui/headless/components/spinner/spinner.qwik.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const qwik = require("@builder.io/qwik");
const spinner = require("./spinner.css.qwik.cjs");
const Spinner = /* @__PURE__ */ qwik.componentQrl(/* @__PURE__ */ qwik.inlinedQrl((props) => {
  qwik.useStylesScopedQrl(/* @__PURE__ */ qwik.inlinedQrl(spinner, "Spinner_component_useStylesScoped_zuLQ9EumCnU"));
  return /* @__PURE__ */ qwik._jsxQ("div", null, {
    class: qwik._fnSignal((p0) => `spinner-container ${p0.class}`, [
      props
    ], "`spinner-container ${p0.class}`"),
    style: qwik._fnSignal((p0) => p0.style, [
      props
    ], "p0.style")
  }, /* @__PURE__ */ qwik._jsxQ("svg", null, {
    class: "spinner",
    part: "base",
    role: "spinner"
  }, [
    /* @__PURE__ */ qwik._jsxQ("circle", null, {
      class: "track",
      r: qwik._fnSignal((p0) => `calc(0.5em - ${p0.width ?? "2px"}/2)`, [
        props
      ], '`calc(0.5em - ${p0.width??"2px"}/2)`'),
      style: qwik._fnSignal((p0) => ({
        strokeWidth: p0.width ?? "2px",
        stroke: p0.trackColor ?? "#8080803f"
      }), [
        props
      ], '{strokeWidth:p0.width??"2px",stroke:p0.trackColor??"#8080803f"}')
    }, null, 3, null),
    /* @__PURE__ */ qwik._jsxQ("circle", null, {
      class: "indicator",
      r: qwik._fnSignal((p0) => `calc(0.5em - ${p0.width ?? "2px"}/2)`, [
        props
      ], '`calc(0.5em - ${p0.width??"2px"}/2)`'),
      style: qwik._fnSignal((p0) => ({
        strokeWidth: p0.width ?? "2px",
        stroke: p0.indicatorColor ?? "#006ce9",
        animation: `spin ${p0.speed ?? "2s"} linear infinite`
      }), [
        props
      ], '{strokeWidth:p0.width??"2px",stroke:p0.indicatorColor??"#006ce9",animation:`spin ${p0.speed??"2s"} linear infinite`}')
    }, null, 3, null)
  ], 3, null), 3, "WQ_0");
}, "Spinner_component_ZwkJC2DeV1w"));
exports.Spinner = Spinner;
