// @qwik-ui/headless/components/spinner/spinner.qwik.mjs
import { componentQrl, inlinedQrl, useStylesScopedQrl, _jsxQ, _fnSignal } from "@builder.io/qwik";
import styles from "./spinner.css.qwik.mjs";
const Spinner = /* @__PURE__ */ componentQrl(/* @__PURE__ */ inlinedQrl((props) => {
  useStylesScopedQrl(/* @__PURE__ */ inlinedQrl(styles, "Spinner_component_useStylesScoped_zuLQ9EumCnU"));
  return /* @__PURE__ */ _jsxQ("div", null, {
    class: _fnSignal((p0) => `spinner-container ${p0.class}`, [
      props
    ], "`spinner-container ${p0.class}`"),
    style: _fnSignal((p0) => p0.style, [
      props
    ], "p0.style")
  }, /* @__PURE__ */ _jsxQ("svg", null, {
    class: "spinner",
    part: "base",
    role: "spinner"
  }, [
    /* @__PURE__ */ _jsxQ("circle", null, {
      class: "track",
      r: _fnSignal((p0) => `calc(0.5em - ${p0.width ?? "2px"}/2)`, [
        props
      ], '`calc(0.5em - ${p0.width??"2px"}/2)`'),
      style: _fnSignal((p0) => ({
        strokeWidth: p0.width ?? "2px",
        stroke: p0.trackColor ?? "#8080803f"
      }), [
        props
      ], '{strokeWidth:p0.width??"2px",stroke:p0.trackColor??"#8080803f"}')
    }, null, 3, null),
    /* @__PURE__ */ _jsxQ("circle", null, {
      class: "indicator",
      r: _fnSignal((p0) => `calc(0.5em - ${p0.width ?? "2px"}/2)`, [
        props
      ], '`calc(0.5em - ${p0.width??"2px"}/2)`'),
      style: _fnSignal((p0) => ({
        strokeWidth: p0.width ?? "2px",
        stroke: p0.indicatorColor ?? "#006ce9",
        animation: `spin ${p0.speed ?? "2s"} linear infinite`
      }), [
        props
      ], '{strokeWidth:p0.width??"2px",stroke:p0.indicatorColor??"#006ce9",animation:`spin ${p0.speed??"2s"} linear infinite`}')
    }, null, 3, null)
  ], 3, null), 3, "WQ_0");
}, "Spinner_component_ZwkJC2DeV1w"));
export {
  Spinner
};
