// @qwik-ui/headless/components/slider/slider.qwik.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const qwik = require("@builder.io/qwik");
const getPercentage = (value, min = 0, max = 100) => {
  return (value - min) * 100 / (max - min);
};
const sliderContext = qwik.createContextId("slider");
const Slider = /* @__PURE__ */ qwik.componentQrl(/* @__PURE__ */ qwik.inlinedQrl((props) => {
  const rootPositionRef = qwik.useSignal();
  const sliderValue = qwik.useSignal(props.value ?? 0);
  const minSignal = qwik.useSignal(props.min ?? 0);
  const maxSignal = qwik.useSignal(props.max ?? 100);
  const positionXSignal = qwik.useSignal();
  const percentageSignal = qwik.useSignal(getPercentage(props.value ?? 0, props.min ?? 0, props.max ?? 100));
  const contextService = {
    value: sliderValue,
    min: minSignal,
    max: maxSignal,
    positionX: positionXSignal,
    percentage: percentageSignal
  };
  qwik.useVisibleTaskQrl(/* @__PURE__ */ qwik.inlinedQrl(async ({ track }) => {
    const [contextService2, rootPositionRef2] = qwik.useLexicalScope();
    track(() => rootPositionRef2);
    contextService2.positionX.value = rootPositionRef2.value?.getBoundingClientRect().x;
  }, "Slider_component_useVisibleTask_NP0ylIsByoU", [
    contextService,
    rootPositionRef
  ]));
  qwik.useVisibleTaskQrl(/* @__PURE__ */ qwik.inlinedQrl(async ({ track }) => {
    const [contextService2, props2, sliderValue2] = qwik.useLexicalScope();
    const newValue = track(() => sliderValue2.value);
    if (props2.onChange$)
      props2.onChange$(newValue);
    contextService2.percentage.value = getPercentage(newValue, props2.min ?? 0, props2.max ?? 100);
  }, "Slider_component_useVisibleTask_1_R0adHmt3YDo", [
    contextService,
    props,
    sliderValue
  ]));
  qwik.useContextProvider(sliderContext, contextService);
  return /* @__PURE__ */ qwik._jsxQ("div", {
    ref: rootPositionRef,
    style: `
          display: inline-block;
          position: relative;
          border: solid 1px rgb(178,178,178);
          border-radius: 4px;
          background: rgb(239,239,239);
          width: 100px;
          height: 6px;
          ${props.style ?? ""}
        `
  }, {
    class: qwik._fnSignal((p0) => p0.class, [
      props
    ], "p0.class")
  }, /* @__PURE__ */ qwik._jsxC(qwik.Slot, null, 3, "aU_0"), 1, "aU_1");
}, "Slider_component_kkWQbL6dvII"));
exports.Slider = Slider;
exports.getPercentage = getPercentage;
exports.sliderContext = sliderContext;
